// @lobehub/ui/es/List/ListItem/style.mjs
import { createStyles } from "antd-style";

//#region src/List/ListItem/style.ts
const useStyles = createStyles(({ cx: cx$1, css: css$1, token, stylish }) => {
	return {
		actions: css$1`
      position: absolute;
      inset-block-start: 50%;
      inset-inline-end: 16px;
      transform: translateY(-50%);
    `,
		active: stylish.active,
		content: css$1`
      position: relative;
      overflow: hidden;
      flex: 1;
      align-self: center;
    `,
		date: css$1`
      font-size: 12px;
      color: ${token.colorTextPlaceholder};
    `,
		desc: css$1`
      width: 100%;
      margin: 0;

      font-size: 12px;
      line-height: 1.2;
      color: ${token.colorTextDescription};
    `,
		pin: css$1`
      position: absolute;
      inset-block-start: 6px;
      inset-inline-end: 6px;
    `,
		root: cx$1(stylish.variantBorderless, css$1`
        cursor: pointer;
        position: relative;
        border-radius: ${token.borderRadius}px;
        color: ${token.colorTextTertiary};
      `),
		title: css$1`
      width: 100%;
      margin: 0;

      font-size: 14px;
      font-weight: 500;
      line-height: 1.2;
      color: ${token.colorText};
    `,
		triangle: css$1`
      width: 10px;
      height: 10px;
      border-radius: 2px;

      opacity: 0.5;
      background: ${token.colorPrimaryBorder};
      clip-path: polygon(0% 0%, 100% 0%, 100% 100%);
    `
	};
});

//#endregion
export { useStyles };
