// @lobehub/ui/es/hooks/useMermaid.mjs
'use client';

import { useMemo, useState } from "react";
import { useTheme } from "antd-style";
import useSWR from "swr";
import { Md5 } from "ts-md5";

//#region src/hooks/useMermaid.ts
const MD5_LENGTH_THRESHOLD = 1e4;
const loadMermaid = () => {
	if (typeof window === "undefined") return Promise.resolve(null);
	return import("mermaid").then((mod) => mod.default);
};
const mermaidPromise = loadMermaid();
/**
* 验证并处理 Mermaid 图表内容
*/
const useMermaid = (content, { id, theme: customTheme }) => {
	const theme = useTheme();
	const [validContent, setValidContent] = useState("");
	const mermaidConfig = useMemo(() => ({
		fontFamily: theme.fontFamilyCode,
		gantt: { useWidth: 1920 },
		securityLevel: "loose",
		startOnLoad: false,
		theme: customTheme || (theme.isDarkMode ? "dark" : "neutral"),
		themeVariables: customTheme ? void 0 : {
			errorBkgColor: theme.colorTextDescription,
			errorTextColor: theme.colorTextDescription,
			fontFamily: theme.fontFamily,
			lineColor: theme.colorTextSecondary,
			mainBkg: theme.colorBgContainer,
			noteBkgColor: theme.colorInfoBg,
			noteTextColor: theme.colorInfoText,
			pie1: theme.geekblue,
			pie2: theme.colorWarning,
			pie3: theme.colorSuccess,
			pie4: theme.colorError,
			primaryBorderColor: theme.colorBorder,
			primaryColor: theme.colorBgContainer,
			primaryTextColor: theme.colorText,
			secondaryBorderColor: theme.colorInfoBorder,
			secondaryColor: theme.colorInfoBg,
			secondaryTextColor: theme.colorInfoText,
			tertiaryBorderColor: theme.colorSuccessBorder,
			tertiaryColor: theme.colorSuccessBg,
			tertiaryTextColor: theme.colorSuccessText,
			textColor: theme.colorText
		}
	}), [theme.isDarkMode, customTheme]);
	return useSWR(useMemo(() => {
		const hash = content.length < MD5_LENGTH_THRESHOLD ? content : Md5.hashStr(content);
		return [
			id,
			customTheme || (theme.isDarkMode ? "d" : "l"),
			hash
		].filter(Boolean).join("-");
	}, [
		content,
		id,
		theme.isDarkMode,
		customTheme
	]), async () => {
		try {
			const mermaidInstance = await mermaidPromise;
			if (!mermaidInstance) return content;
			if (await mermaidInstance.parse(content)) {
				mermaidInstance.initialize(mermaidConfig);
				const { svg } = await mermaidInstance.render(id, content);
				setValidContent(svg);
				return svg;
			} else throw new Error("Mermaid 语法无效");
		} catch (error) {
			if (!validContent) console.error("Mermaid 解析错误:", error);
			return validContent || "";
		}
	}, {
		dedupingInterval: 3e3,
		errorRetryCount: 2,
		revalidateOnFocus: false,
		revalidateOnReconnect: false
	});
};

//#endregion
export { useMermaid };
