// @lobehub/ui/es/mdx/FileTree/Folder.mjs
'use client';

import FlexBasic_default from "../../Flex/FlexBasic.mjs";
import Icon_default from "../../Icon/Icon.mjs";
import { useState } from "react";
import { jsx, jsxs } from "react/jsx-runtime";
import { createStyles } from "antd-style";
import { FolderIcon, FolderOpen } from "lucide-react";

//#region src/mdx/FileTree/Folder.tsx
const useStyles = createStyles(({ css: css$1, token }) => {
	return {
		folder: css$1`
      cursor: pointer;

      &:hover {
        color: ${token.colorText};
      }
    `,
		folderChildren: css$1`
      padding-inline-start: 1em;
    `
	};
});
const Folder = ({ name, defaultOpen, icon = FolderIcon, children, ...rest }) => {
	const [open, setOpen] = useState(defaultOpen);
	const { styles } = useStyles();
	return /* @__PURE__ */ jsxs(FlexBasic_default, {
		...rest,
		children: [/* @__PURE__ */ jsxs(FlexBasic_default, {
			align: "center",
			className: styles.folder,
			gap: 4,
			horizontal: true,
			onClick: () => setOpen(!open),
			children: [/* @__PURE__ */ jsx(Icon_default, { icon: open ? FolderOpen : icon }), /* @__PURE__ */ jsx("span", { children: name })]
		}), open && /* @__PURE__ */ jsx(FlexBasic_default, {
			className: styles.folderChildren,
			children
		})]
	});
};
Folder.displayName = "MdxFolder";
var Folder_default = Folder;

//#endregion
export { Folder_default as default };
