// @lobehub/ui/es/Modal/style.mjs
import { createStyles } from "antd-style";

//#region src/Modal/style.ts
const HEADER_HEIGHT = 56;
const useStyles = createStyles(({ css: css$1, token, prefixCls }) => {
	return {
		content: css$1`
      .${prefixCls}-modal-footer {
        margin: 0;
        padding: 16px;
      }
      .${prefixCls}-modal-header {
        display: flex;
        gap: 4px;
        align-items: center;
        justify-content: center;

        height: 56px;
        margin-block-end: 0;
        padding: 16px;
      }
      .${prefixCls}-modal-container {
        overflow: hidden;
        padding: 0;
        border: 1px solid ${token.colorSplit};
        border-radius: ${token.borderRadiusLG}px;
      }
    `,
		drawerContent: css$1`
      .${prefixCls}-drawer-close {
        padding: 0;
      }

      .${prefixCls}-drawer-header {
        flex: none;
        height: ${HEADER_HEIGHT}px !important;
        padding-block: 0;
        padding-inline: 16px;
      }

      .${prefixCls}-drawer-footer {
        display: flex;
        align-items: center;
        justify-content: flex-end;

        padding: 16px;
        border: none;
      }
    `,
		wrap: css$1`
      overflow: hidden auto;
    `
	};
});

//#endregion
export { useStyles };
