// @lobehub/ui/es/Input/Input.mjs
'use client';

import { useStyles } from "./style.mjs";
import { memo, useMemo } from "react";
import { jsx } from "react/jsx-runtime";
import { Input } from "antd";
import { cva } from "class-variance-authority";

//#region src/Input/Input.tsx
const Input$1 = memo(({ ref, variant, shadow, className, ...rest }) => {
	const { styles, cx, theme } = useStyles();
	return /* @__PURE__ */ jsx(Input, {
		className: cx(useMemo(() => cva(styles.root, {
			defaultVariants: { shadow: false },
			variants: {
				variant: {
					filled: styles.filled,
					outlined: styles.outlined,
					borderless: styles.borderless,
					underlined: null
				},
				shadow: {
					false: null,
					true: styles.shadow
				}
			}
		}), [styles])({
			shadow,
			variant: variant || (theme.isDarkMode ? "filled" : "outlined")
		}), className),
		ref,
		variant: variant || (theme.isDarkMode ? "filled" : "outlined"),
		...rest
	});
});
Input$1.displayName = "Input";
var Input_default = Input$1;

//#endregion
export { Input_default as default };
