// @lobehub/ui/es/Input/InputPassword.mjs
'use client';

import { useStyles } from "./style.mjs";
import { memo, useMemo } from "react";
import { jsx } from "react/jsx-runtime";
import { Input } from "antd";
import { cva } from "class-variance-authority";

//#region src/Input/InputPassword.tsx
const InputPassword = memo(({ ref, variant, shadow, className, ...rest }) => {
	const { styles, cx, theme } = useStyles();
	const variants = useMemo(() => cva(styles.root, {
		defaultVariants: { shadow: false },
		variants: {
			variant: {
				filled: styles.filled,
				outlined: styles.outlined,
				borderless: styles.borderless,
				underlined: null
			},
			shadow: {
				false: null,
				true: styles.shadow
			}
		}
	}), [styles]);
	return /* @__PURE__ */ jsx(Input.Password, {
		className: cx(variants({
			shadow,
			variant: variant || (theme.isDarkMode ? "filled" : "outlined")
		}), className),
		ref,
		variant: variant || (theme.isDarkMode ? "filled" : "outlined"),
		...rest
	});
});
InputPassword.displayName = "InputPassword";
var InputPassword_default = InputPassword;

//#endregion
export { InputPassword_default as default };
