// @lobehub/ui/es/Input/InputOPT.mjs
'use client';

import { useStyles } from "./style.mjs";
import { memo, useMemo } from "react";
import { jsx } from "react/jsx-runtime";
import { Input } from "antd";
import { cva } from "class-variance-authority";

//#region src/Input/InputOPT.tsx
const InputOPT = memo(({ ref, variant, shadow, className, ...rest }) => {
	const { styles, cx, theme } = useStyles();
	const variants = useMemo(() => cva(styles.rootOPT, {
		defaultVariants: { shadow: false },
		variants: {
			variant: {
				filled: styles.filledOPT,
				outlined: styles.outlinedOPT,
				borderless: styles.borderlessOPT,
				underlined: null
			},
			shadow: {
				false: null,
				true: styles.shadowOPT
			}
		}
	}), [styles]);
	return /* @__PURE__ */ jsx(Input.OTP, {
		className: cx(variants({
			shadow,
			variant: variant || (theme.isDarkMode ? "filled" : "outlined")
		}), className),
		ref,
		variant: variant || (theme.isDarkMode ? "filled" : "outlined"),
		...rest
	});
});
InputOPT.displayName = "InputOPT";
var InputOPT_default = InputOPT;

//#endregion
export { InputOPT_default as default };
