// @lobehub/ui/es/Input/style.mjs
import { createStyles } from "antd-style";

//#region src/Input/style.ts
const useStyles = createStyles(({ cx: cx$1, token, css: css$1, stylish, prefixCls }) => {
	return {
		borderless: css$1`
      &.${prefixCls}-input {
        ${stylish.variantBorderless}
        &:hover {
          ${stylish.variantBorderlessWithoutHover}
        }
      }
    `,
		borderlessOPT: css$1`
      &.${prefixCls}-otp {
        .${prefixCls}-otp-input {
          ${stylish.variantBorderless};
        }
      }
    `,
		filled: cx$1(stylish.variantFilled, css$1`
        &:focus-within {
          ${stylish.variantFilledWithoutHover}
        }
      `),
		filledOPT: css$1`
      &.${prefixCls}-otp {
        .${prefixCls}-otp-input {
          ${stylish.variantFilled};
        }
      }
    `,
		outlined: stylish.variantOutlined,
		outlinedOPT: css$1`
      &.${prefixCls}-otp {
        .${prefixCls}-otp-input {
          ${stylish.variantOutlined};
        }
      }
    `,
		root: css$1``,
		rootOPT: css$1`
      &.${prefixCls}-otp {
        .${prefixCls}-otp-input {
          &:focus-within {
            border-color: ${token.colorBorder};
          }
        }
      }
    `,
		shadow: stylish.shadow,
		shadowOPT: css$1`
      &.${prefixCls}-otp {
        .${prefixCls}-otp-input {
          ${stylish.shadow};
        }
      }
    `
	};
});

//#endregion
export { useStyles };
