// @lobehub/ui/es/Input/TextArea.mjs
'use client';

import { useStyles } from "./style.mjs";
import { memo, useMemo } from "react";
import { jsx } from "react/jsx-runtime";
import { Input } from "antd";
import { cva } from "class-variance-authority";

//#region src/Input/TextArea.tsx
const TextArea = memo(({ ref, variant, shadow, className, resize = false, style, ...rest }) => {
	const { styles, cx, theme } = useStyles();
	const variants = useMemo(() => cva(styles.root, {
		defaultVariants: { shadow: false },
		variants: {
			variant: {
				filled: styles.filled,
				outlined: styles.outlined,
				borderless: styles.borderless,
				underlined: null
			},
			shadow: {
				false: null,
				true: styles.shadow
			}
		}
	}), [styles]);
	return /* @__PURE__ */ jsx(Input.TextArea, {
		className: cx(variants({
			shadow,
			variant: variant || (theme.isDarkMode ? "filled" : "outlined")
		}), className),
		ref,
		style: {
			resize: resize ? void 0 : "none",
			...style
		},
		variant: variant || (theme.isDarkMode ? "filled" : "outlined"),
		...rest
	});
});
TextArea.displayName = "TextArea";
var TextArea_default = TextArea;

//#endregion
export { TextArea_default as default };
