// @lobehub/ui/es/Burger/Burger.mjs
'use client';

import Center_default from "../Flex/Center.mjs";
import ActionIcon_default from "../ActionIcon/ActionIcon.mjs";
import { useStyles } from "./style.mjs";
import { memo } from "react";
import { jsx, jsxs } from "react/jsx-runtime";
import { Drawer, Menu } from "antd";
import { MenuIcon, X } from "lucide-react";

//#region src/Burger/Burger.tsx
const Burger = memo(({ items, openKeys, selectedKeys, opened, setOpened, className, headerHeight = 64, onClick, iconProps, size, variant, rootClassName, fullscreen, drawerProps, ...rest }) => {
	const { cx, styles } = useStyles({
		fullscreen,
		headerHeight
	});
	return /* @__PURE__ */ jsxs(Center_default, {
		className: cx(styles.container, className),
		onClick: () => {
			setOpened(!opened);
		},
		...rest,
		children: [/* @__PURE__ */ jsx(ActionIcon_default, {
			icon: opened ? X : MenuIcon,
			size,
			variant,
			...iconProps
		}), /* @__PURE__ */ jsxs(Drawer, {
			closeIcon: void 0,
			open: opened,
			placement: "left",
			width: "100vw",
			...drawerProps,
			className: styles.drawer,
			rootClassName: cx(styles.drawerRoot, rootClassName),
			styles: {
				body: { padding: 0 },
				header: { display: "none" }
			},
			children: [/* @__PURE__ */ jsx(Menu, {
				className: styles.menu,
				items,
				mode: "inline",
				onClick,
				openKeys,
				selectedKeys
			}), /* @__PURE__ */ jsx("div", { className: styles.fillRect })]
		})]
	});
});
Burger.displayName = "Burger";
var Burger_default = Burger;

//#endregion
export { Burger_default as default };
