// @lobehub/ui/es/Block/Block.mjs
'use client';

import FlexBasic_default from "../Flex/FlexBasic.mjs";
import { useStyles } from "./style.mjs";
import { useMemo } from "react";
import { jsx } from "react/jsx-runtime";
import { cva } from "class-variance-authority";

//#region src/Block/Block.tsx
const Block = ({ className, variant = "filled", shadow, glass, children, clickable, ref, ...rest }) => {
	const { cx, styles } = useStyles();
	return /* @__PURE__ */ jsx(FlexBasic_default, {
		className: cx(useMemo(() => cva(styles.root, {
			compoundVariants: [
				{
					class: styles.clickableBorderless,
					clickable: true,
					variant: "borderless"
				},
				{
					class: styles.clickableFilled,
					clickable: true,
					variant: "filled"
				},
				{
					class: styles.clickableOutlined,
					clickable: true,
					variant: "outlined"
				}
			],
			defaultVariants: {
				clickable: false,
				glass: false,
				shadow: false,
				variant: "filled"
			},
			variants: {
				variant: {
					filled: styles.filled,
					outlined: styles.outlined,
					borderless: styles.borderless
				},
				clickable: {
					false: null,
					true: styles.clickableRoot
				},
				glass: {
					false: null,
					true: styles.glass
				},
				shadow: {
					false: null,
					true: styles.shadow
				}
			}
		}), [styles])({
			clickable,
			glass,
			shadow,
			variant
		}), className),
		ref,
		...rest,
		children
	});
};
Block.displayName = "Block";
var Block_default = Block;

//#endregion
export { Block_default as default };
