// @lobehub/ui/es/Menu/style.mjs
import { createStyles } from "antd-style";

//#region src/Menu/style.ts
const useStyles = createStyles(({ cx: cx$1, css: css$1, token, stylish, prefixCls }) => ({
	borderless: cx$1(stylish.variantBorderlessWithoutHover, css$1`
      padding: 0;
      border-radius: unset;
    `),
	compact: css$1`
    &.${prefixCls}-menu {
      .${prefixCls}-menu-item-divider {
        margin: 0;
      }
    }
  `,
	filled: stylish.variantFilledWithoutHover,
	outlined: stylish.variantOutlinedWithoutHover,
	root: css$1`
    &.${prefixCls}-menu {
      flex: 1;

      padding: 4px;

      background: transparent;
      border: none !important;
      border-radius: ${token.borderRadiusLG}px;

      .${prefixCls}-menu-sub.${prefixCls}-menu-inline {
        background: transparent;
        > .${prefixCls}-menu-item {
          padding-inline-start: 36px !important;
        }
      }

      .${prefixCls}-menu-item-divider {
        margin-block: 1em;
      }
    }
  `,
	shadow: stylish.shadow
}));

//#endregion
export { useStyles };
