// @lobehub/ui/es/Collapse/style.mjs
import { createStyles } from "antd-style";

//#region src/Collapse/style.ts
const DEFAULT_PADDING = "12px 16px";
const getPadding = (padding) => !padding && padding !== 0 ? DEFAULT_PADDING : `${typeof padding === "string" ? padding : `${padding}px`} !important`;
const useStyles = createStyles(({ css: css$1, token, prefixCls, stylish, isDarkMode }) => {
	return {
		borderless: css$1`
      &.${prefixCls}-collapse {
        .${prefixCls}-collapse-header {
          padding-inline: 0;
        }
        .${prefixCls}-collapse-content {
          padding-inline: 0;
          .${prefixCls}-collapse-content-box {
            padding-inline: 0;
          }
        }
      }
    `,
		desc: css$1`
      font-size: 12px;
      color: ${token.colorTextDescription};
    `,
		filled: css$1`
      &.${prefixCls}-collapse {
        .${prefixCls}-collapse-item {
          background: ${isDarkMode ? token.colorFillQuaternary : token.colorFillTertiary};
          .${prefixCls}-collapse-content {
            margin-inline: 3px;
            margin-block-end: 3px;
            border-radius: ${token.borderRadius}px;
            ${stylish.variantOutlinedWithoutHover};
            background: ${isDarkMode ? token.colorFillQuaternary : token.colorBgContainer};
            ${isDarkMode ? void 0 : stylish.shadow};
          }
        }
      }
    `,
		gapOutlined: css$1`
      &.${prefixCls}-collapse {
        background: transparent;
        .${prefixCls}-collapse-item {
          border: 1px solid ${token.colorFillSecondary};
          background: ${token.colorBgContainer};
        }

        .${prefixCls}-collapse-item:not(:first-child) {
          .${prefixCls}-collapse-header {
            border-block-start: none;
          }
        }
      }
    `,
		gapRoot: css$1`
      &.${prefixCls}-collapse {
        display: flex;
        flex-direction: column;
        border: none;
        box-shadow: none;
        .${prefixCls}-collapse-item {
          border: none;
          border-radius: ${token.borderRadiusLG}px;
        }
      }
    `,
		hideCollapsibleIcon: css$1`
      .${prefixCls}-collapse-expand-icon {
        display: none !important;
      }
    `,
		icon: css$1`
      cursor: pointer;
      transition: all 100ms ${token.motionEaseOut};
    `,
		outlined: css$1`
      &.${prefixCls}-collapse {
        border: 1px solid ${token.colorFillSecondary};
        background: ${token.colorBgContainer};
        .${prefixCls}-collapse-item .${prefixCls}-collapse-header {
          transition: none;
        }
        .${prefixCls}-collapse-item-active .${prefixCls}-collapse-header {
          border-block-end: 1px solid ${token.colorFillTertiary};
        }
        .${prefixCls}-collapse-item:not(:first-child) {
          .${prefixCls}-collapse-header {
            border-block-start: 1px solid ${token.colorFillTertiary};
          }
        }
      }
    `,
		root: css$1`
      &.${prefixCls}-collapse {
        display: flex;
        flex-direction: column;
        background: transparent;

        .${prefixCls}-collapse-header {
          overflow: hidden;
          display: flex;
          flex: none;
          gap: 0.75em;
          align-items: flex-start;

          border-radius: 0 !important;

          .${prefixCls}-collapse-header-text {
            flex: 1;
          }

          .${prefixCls}-collapse-expand-icon {
            align-items: center;
            min-height: 28px;
            padding: 0;
          }

          .${prefixCls}-collapse-extra {
            display: flex;
            align-items: center;
            min-height: 28px;
          }
        }

        .${prefixCls}-collapse-content {
          background: transparent;
        }
      }
    `,
		title: css$1`
      font-size: 16px;
      font-weight: 500;
      line-height: 28px;
    `
	};
});

//#endregion
export { DEFAULT_PADDING, getPadding, useStyles };
