// @lobehub/ui/es/Button/Button.mjs
'use client';

import Icon_default from "../Icon/Icon.mjs";
import { useStyles } from "./style.mjs";
import { isValidElement, useMemo } from "react";
import { jsx } from "react/jsx-runtime";
import { Button } from "antd";
import { cva } from "class-variance-authority";
import { Loader2Icon } from "lucide-react";

//#region src/Button/Button.tsx
const Button$1 = ({ icon, variant, glass, shadow, loading, className, type, color, danger, children, iconProps, ref, ...rest }) => {
	const { styles, cx, theme } = useStyles();
	const defaultVariant = type ? void 0 : variant || (theme.isDarkMode ? "filled" : "outlined");
	return /* @__PURE__ */ jsx(Button, {
		className: cx(useMemo(() => cva(styles.root, {
			defaultVariants: {
				glass: false,
				shadow: false
			},
			variants: {
				glass: {
					false: null,
					true: styles.glass
				},
				shadow: {
					false: null,
					true: styles.shadow
				}
			}
		}), [styles])({
			glass,
			shadow
		}), className),
		color: color || (defaultVariant === "filled" ? "default" : void 0),
		danger,
		icon: icon && (isValidElement(icon) ? icon : /* @__PURE__ */ jsx(Icon_default, {
			icon,
			...iconProps,
			size: iconProps?.size || { size: "1.2em" }
		})),
		loading: loading ? { icon: /* @__PURE__ */ jsx(Icon_default, {
			icon: Loader2Icon,
			...iconProps,
			size: iconProps?.size || { size: "1.2em" },
			spin: true
		}) } : false,
		ref,
		type,
		variant: defaultVariant,
		...rest,
		children
	});
};
var Button_default = Button$1;

//#endregion
export { Button_default as default };
