// @lobehub/ui/es/ThemeProvider/ConfigProvider.mjs
'use client';

import { memo } from "react";
import { jsx } from "react/jsx-runtime";
import { ConfigProvider } from "antd";
import { useTheme } from "antd-style";

//#region src/ThemeProvider/ConfigProvider.tsx
const ConfigProvider$1 = memo(({ children }) => {
	const theme = useTheme();
	return /* @__PURE__ */ jsx(ConfigProvider, {
		theme: { components: {
			Button: { contentFontSizeSM: 12 },
			DatePicker: {
				activeBorderColor: theme.colorBorder,
				hoverBorderColor: theme.colorBorder
			},
			Input: {
				activeBorderColor: theme.colorBorder,
				hoverBorderColor: theme.colorBorder
			},
			InputNumber: {
				activeBorderColor: theme.colorBorder,
				hoverBorderColor: theme.colorBorder
			},
			Mentions: {
				activeBorderColor: theme.colorBorder,
				hoverBorderColor: theme.colorBorder
			},
			Select: {
				activeBorderColor: theme.colorBorder,
				hoverBorderColor: theme.colorBorder
			}
		} },
		children
	});
});
var ConfigProvider_default = ConfigProvider$1;

//#endregion
export { ConfigProvider_default as default };
