// @lobehub/ui/es/ScrollShadow/style.mjs
import { createStyles } from "antd-style";

//#region src/ScrollShadow/style.ts
const useStyles = createStyles(({ css: css$1 }, size) => {
	const shadowSize = size + "%";
	return {
		bottomShadow: css$1`
      mask-image: linear-gradient(180deg, #000 calc(100% - ${shadowSize}), transparent);
    `,
		hideScrollBar: css$1`
      scrollbar-width: none;

      -ms-overflow-style: none;

      &::-webkit-scrollbar {
        display: none;
      }
    `,
		horizontal: css$1`
      overflow-x: auto;
    `,
		leftRightShadow: css$1`
      mask-image: linear-gradient(
        to right,
        #000,
        #000,
        transparent 0,
        #000 ${shadowSize},
        #000 calc(100% - ${shadowSize}),
        transparent
      );
    `,
		leftShadow: css$1`
      mask-image: linear-gradient(270deg, #000 calc(100% - ${shadowSize}), transparent);
    `,
		rightShadow: css$1`
      mask-image: linear-gradient(90deg, #000 calc(100% - ${shadowSize}), transparent);
    `,
		root: css$1`
      position: relative;
      overflow: hidden;
    `,
		topBottomShadow: css$1`
      mask-image: linear-gradient(
        #000,
        #000,
        transparent 0,
        #000 ${shadowSize},
        #000 calc(100% - ${shadowSize}),
        transparent
      );
    `,
		topShadow: css$1`
      mask-image: linear-gradient(0deg, #000 calc(100% - ${shadowSize}), transparent);
    `,
		vertical: css$1`
      overflow-y: auto;
    `
	};
});

//#endregion
export { useStyles };
