// @lobehub/ui/es/Alert/style.mjs
import { createStyles } from "antd-style";

//#region src/Alert/style.ts
const useStyles = createStyles(({ cx: cx$1, token, css: css$1, prefixCls, stylish }, { closable, hasTitle, showIcon }) => {
	const baseBlockPadding = hasTitle ? 16 : 8;
	const baseInlinePadding = hasTitle ? 16 : 12;
	return {
		banner: css$1`
        border: none !important;
        border-radius: 0 !important;
      `,
		borderless: css$1`
        padding: 0 !important;
        border: none !important;
        background: transparent !important;
      `,
		borderlessExtraHeader: css$1`
        margin-block-start: ${baseBlockPadding}px;
        padding-inline: 0;
      `,
		colorfulText: css$1`
        .${prefixCls}-alert-message,.${prefixCls}-alert-description {
          color: inherit;
        }
      `,
		expandText: css$1`
        padding-inline-end: 12px;

        &:hover {
          cursor: pointer;
        }
      `,
		extra: css$1`
        position: relative;

        overflow: hidden;

        max-width: 100%;
        border: 1px solid;
        border-block-start: none;
        border-end-start-radius: ${token.borderRadiusLG}px;
        border-end-end-radius: ${token.borderRadiusLG}px;
      `,
		extraHeader: css$1`
        border-block-start: 1px dashed;
        border-radius: 0;
        background: transparent !important;
      `,
		filled: css$1``,
		glass: stylish.blur,
		hasExtra: css$1`
        border-block-end: none;
        border-end-start-radius: 0;
        border-end-end-radius: 0;
      `,
		outlined: css$1`
        background: transparent !important;
      `,
		root: cx$1(css$1`
          position: relative;

          display: flex;
          flex-direction: row;
          gap: ${hasTitle ? 12 : 8}px;
          align-items: flex-start;

          max-width: 100%;
          padding-block: ${baseBlockPadding}px;
          padding-inline: ${showIcon ? baseInlinePadding * .75 : baseInlinePadding}px
            ${closable ? baseInlinePadding * .75 : baseInlinePadding}px;

          .${prefixCls}-alert-title {
            font-weight: ${hasTitle ? 500 : 400};
            line-height: 24px;
            color: inherit;
            word-break: normal;
          }
          .${prefixCls}-alert-icon {
            display: flex;
            align-items: center;
            height: 24px;
            margin: 0;
          }
          .${prefixCls}-alert-close-icon {
            display: flex;
            align-items: center;
            height: 24px;
            margin: 0;
          }
        `, hasTitle && css$1`
            .${prefixCls}-alert-description {
              line-height: 1.5;
              word-break: normal;
              opacity: 0.75;
            }
          `)
	};
});

//#endregion
export { useStyles };
