// @lobehub/ui/es/Toc/style.mjs
import { createStyles } from "antd-style";
import { rgba } from "polished";

//#region src/Toc/style.ts
const useStyles = createStyles(({ token, stylish, responsive, cx: cx$1, css: css$1, prefixCls }, { tocWidth, headerHeight }) => {
	return {
		anchor: cx$1(stylish.blur, css$1`
          overflow: hidden auto;
          max-height: 60dvh !important;
        `),
		container: css$1`
        position: sticky;
        inset-block-start: ${headerHeight + 64}px;

        overscroll-behavior: contain;
        grid-area: toc;

        width: ${tocWidth}px;
        margin-inline-end: 24px;

        border-radius: 3px;

        -webkit-overflow-scrolling: touch;

        ${responsive.mobile} {
          position: relative;
          inset-inline-start: 0;
          width: 100%;
          margin-block-start: 0;
        }

        > h4 {
          margin-block: 0 8px;
          margin-inline: 0;

          font-size: 12px;
          line-height: 1;
          color: ${token.colorTextDescription};
        }
      `,
		expand: cx$1(stylish.blur, css$1`
          width: 100%;

          background-color: ${rgba(token.colorBgLayout, .5)};
          border-block-end: 1px solid ${token.colorSplit};
          border-radius: 0;
          box-shadow: ${token.boxShadowSecondary};

          .${prefixCls}-collapse-content {
            overflow: auto;
          }

          .${prefixCls}-collapse-header {
            z-index: 10;
            padding-block: 8px !important;
            padding-inline: 16px 8px !important;
          }
        `),
		mobileCtn: css$1`
        width: 100%;
        height: ${36}px;

        .${prefixCls}-collapse-expand-icon {
          color: ${token.colorTextQuaternary};
        }
      `
	};
});

//#endregion
export { useStyles };
