// @lobehub/ui/es/Video/index.mjs
'use client';

import FlexBasic_default from "../Flex/FlexBasic.mjs";
import ActionIcon_default from "../ActionIcon/ActionIcon.mjs";
import { useStyles } from "./style.mjs";
import { memo, useMemo, useState } from "react";
import { jsx, jsxs } from "react/jsx-runtime";
import { Skeleton } from "antd";
import { cva } from "class-variance-authority";
import { PlayIcon } from "lucide-react";

//#region src/Video/index.tsx
const Video = memo(({ ref, preload = "auto", src, style, classNames, className, maxHeight = "100%", maxWidth = "100%", minHeight, minWidth, onEnded, onPause, onPlay, onPlaying, width, height, onMouseEnter, styles: customStyles, onMouseLeave, preview = true, isLoading, variant = "filled", autoPlay, ...rest }) => {
	const [isPlaying, setIsPlaying] = useState(false);
	const [showControls, setShowControls] = useState(false);
	const { cx, styles, theme } = useStyles({
		maxHeight,
		maxWidth,
		minHeight,
		minWidth
	});
	const variants = useMemo(() => cva(styles.root, {
		defaultVariants: { variant: "filled" },
		variants: { variant: {
			filled: styles.filled,
			outlined: styles.outlined,
			borderless: styles.borderless
		} }
	}), [styles]);
	if (isLoading) return /* @__PURE__ */ jsx(Skeleton.Avatar, {
		active: true,
		style: {
			borderRadius: theme.borderRadius,
			height,
			maxHeight,
			maxWidth,
			minHeight,
			minWidth,
			width
		}
	});
	return /* @__PURE__ */ jsxs(FlexBasic_default, {
		className: cx(variants({ variant }), className, classNames?.wrapper),
		height,
		ref,
		style: {
			...style,
			...customStyles?.wrapper
		},
		width,
		children: [preview && !isPlaying && /* @__PURE__ */ jsx(FlexBasic_default, {
			align: "center",
			className: cx(styles.mask, classNames?.mask),
			justify: "center",
			style: customStyles?.mask,
			children: /* @__PURE__ */ jsx(ActionIcon_default, {
				color: "#fff",
				icon: PlayIcon,
				variant: "filled"
			})
		}), /* @__PURE__ */ jsx("video", {
			autoPlay,
			className: cx(styles.video, classNames?.video),
			controls: showControls,
			height,
			onEnded: (e) => {
				setIsPlaying(false);
				onEnded?.(e);
			},
			onMouseEnter: (e) => {
				setShowControls(true);
				onMouseEnter?.(e);
			},
			onMouseLeave: (e) => {
				setShowControls(false);
				onMouseLeave?.(e);
			},
			onPause: (e) => {
				setIsPlaying(false);
				onPause?.(e);
			},
			onPlay: (e) => {
				setIsPlaying(true);
				onPlay?.(e);
			},
			onPlaying: (e) => {
				setIsPlaying(true);
				onPlaying?.(e);
			},
			preload,
			style: {
				height: "auto",
				maxWidth: "100%",
				...customStyles?.video
			},
			width,
			...rest,
			children: /* @__PURE__ */ jsx("source", { src })
		})]
	});
});
Video.displayName = "Video";
var Video_default = Video;

//#endregion
export { Video_default as default };
