// @lobehub/ui/es/Video/style.mjs
import { createStyles } from "antd-style";

//#region src/Video/style.ts
const useStyles = createStyles(({ css: css$1, token, cx: cx$1, stylish }, { maxHeight, maxWidth, minWidth, minHeight } = {}) => {
	const MAX_HEIGHT = typeof maxHeight === "number" ? `${maxHeight}px` : maxHeight;
	const MAX_WIDTH = typeof maxWidth === "number" ? `${maxWidth}px` : maxWidth;
	const MIN_HEIGHT = typeof minHeight === "number" ? `${minHeight}px` : minHeight;
	const MIN_WIDTH = typeof minWidth === "number" ? `${minWidth}px` : minWidth;
	const mask = cx$1(css$1`
      pointer-events: none;

      position: absolute;
      z-index: 1;
      inset: 0;

      width: 100%;
      height: 100%;

      opacity: 0;
      background: ${token.colorBgMask};

      transition: opacity 0.3s;
    `);
	return {
		borderless: stylish.variantBorderlessWithoutHover,
		filled: cx$1(stylish.variantOutlinedWithoutHover, stylish.variantFilledWithoutHover),
		mask,
		outlined: stylish.variantOutlinedWithoutHover,
		root: css$1`
        position: relative;

        overflow: hidden;

        width: 100%;
        min-width: ${MIN_WIDTH};
        max-width: ${MAX_WIDTH};
        height: auto;
        min-height: ${MIN_HEIGHT};
        max-height: ${MAX_HEIGHT};
        margin-block: 0 1em;

        background: ${token.colorFillTertiary};
        border-radius: ${token.borderRadius}px;

        &:hover {
          .${mask} {
            opacity: 1;
          }
        }
      `,
		video: css$1`
        cursor: pointer;
        width: 100%;
      `
	};
});

//#endregion
export { useStyles };
