// @lobehub/ui/es/Text/styles.mjs
import { createStyles } from "antd-style";

//#region src/Text/styles.ts
const useStyles = createStyles(({ css: css$1, token }) => ({
	code: css$1`
    font-family: ${token.fontFamilyCode};
  `,
	danger: css$1`
    color: ${token.colorError};
  `,
	delete: css$1`
    text-decoration: line-through;
  `,
	disabled: css$1`
    cursor: not-allowed;
    color: ${token.colorTextDisabled};
  `,
	ellipsis: css$1`
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;
  `,
	ellipsisMulti: css$1`
    overflow: hidden;
    display: -webkit-box;
    -webkit-box-orient: vertical;
    text-overflow: ellipsis;
  `,
	h1: css$1`
    font-size: calc(${token.fontSize}px * 2.5);
    font-weight: bold;
    line-height: 1.25;
  `,
	h2: css$1`
    font-size: calc(${token.fontSize}px * 2);
    font-weight: bold;
    line-height: 1.25;
  `,
	h3: css$1`
    font-size: calc(${token.fontSize}px * 1.5);
    font-weight: bold;
    line-height: 1.25;
  `,
	h4: css$1`
    font-size: calc(${token.fontSize}px * 1.25);
    font-weight: bold;
    line-height: 1.25;
  `,
	h5: css$1`
    font-size: ${token.fontSize}px;
    font-weight: bold;
    line-height: 1.25;
  `,
	info: css$1`
    color: ${token.colorInfo};
  `,
	italic: css$1`
    font-style: italic;
  `,
	mark: css$1`
    color: #000;
    background-color: ${token.yellow};
  `,
	p: css$1`
    margin-block: 0;
  `,
	secondary: css$1`
    color: ${token.colorTextDescription};
  `,
	strong: css$1`
    font-weight: bold;
  `,
	success: css$1`
    color: ${token.colorSuccess};
  `,
	text: css$1`
    color: ${token.colorText};
  `,
	underline: css$1`
    text-decoration: underline;
  `,
	warning: css$1`
    color: ${token.colorWarning};
  `
}));

//#endregion
export { useStyles };
