// @lobehub/ui/es/Text/Text.mjs
'use client';

import Tooltip_default from "../Tooltip/Tooltip.mjs";
import { useStyles } from "./styles.mjs";
import { useMemo } from "react";
import { jsx } from "react/jsx-runtime";
import { cva } from "class-variance-authority";

//#region src/Text/Text.tsx
const Text = ({ as: Container = "div", align, className, children, style, type, disabled, strong, italic, underline, delete: deleteStyle, fontSize, mark, code, color, weight, ellipsis, ...rest }) => {
	const { styles, cx } = useStyles();
	const variants = useMemo(() => cva(styles.text, {
		defaultVariants: {},
		variants: {
			as: {
				h1: styles.h1,
				h2: styles.h2,
				h3: styles.h3,
				h4: styles.h4,
				h5: styles.h5,
				p: styles.p
			},
			code: { true: styles.code },
			delete: { true: styles.delete },
			disabled: { true: styles.disabled },
			ellipsis: {
				multi: styles.ellipsisMulti,
				true: styles.ellipsis
			},
			italic: { true: styles.italic },
			mark: { true: styles.mark },
			strong: { true: styles.strong },
			type: {
				danger: styles.danger,
				info: styles.info,
				secondary: styles.secondary,
				success: styles.success,
				warning: styles.warning
			},
			underline: { true: styles.underline }
		}
	}), [styles]);
	const textStyle = {
		...color && { color },
		...weight && { fontWeight: weight },
		...typeof ellipsis === "object" && ellipsis.rows && { WebkitLineClamp: ellipsis.rows },
		...fontSize && { fontSize },
		...align && { textAlign: align },
		...style
	};
	const content = /* @__PURE__ */ jsx(Container, {
		className: cx(variants({
			as: [
				"h1",
				"h2",
				"h3",
				"h4",
				"h5",
				"p"
			].includes(Container) ? Container : void 0,
			code,
			delete: deleteStyle,
			disabled,
			ellipsis: ellipsis ? typeof ellipsis === "object" && ellipsis.rows ? "multi" : true : void 0,
			italic,
			mark,
			strong,
			type,
			underline
		}), className),
		style: textStyle,
		...rest,
		children
	});
	if (ellipsis && typeof ellipsis === "object" && ellipsis.tooltip) {
		const title = typeof ellipsis.tooltip === "string" ? ellipsis.tooltip : children;
		if (typeof ellipsis.tooltip === "object") return /* @__PURE__ */ jsx(Tooltip_default, {
			...ellipsis.tooltip,
			title: ellipsis.tooltip?.title || title,
			children: content
		});
		return /* @__PURE__ */ jsx(Tooltip_default, {
			title,
			children: content
		});
	}
	return content;
};
Text.displayName = "Text";
var Text_default = Text;

//#endregion
export { Text_default as default };
