// @lobehub/ui/es/Tabs/Tabs.mjs
'use client';

import ActionIcon_default from "../ActionIcon/ActionIcon.mjs";
import { useStyles } from "./style.mjs";
import { useMemo } from "react";
import { jsx } from "react/jsx-runtime";
import { Tabs } from "antd";
import { cva } from "class-variance-authority";
import { MoreHorizontalIcon } from "lucide-react";

//#region src/Tabs/Tabs.tsx
const Tabs$1 = ({ className, compact, variant = "rounded", items, ...rest }) => {
	const { styles, cx } = useStyles();
	const hasContent = items?.some((item) => !!item.children);
	return /* @__PURE__ */ jsx(Tabs, {
		className: cx(useMemo(() => cva(styles.root, {
			defaultVariants: {
				compact: false,
				underlined: false,
				variant: "rounded"
			},
			variants: {
				variant: {
					square: null,
					rounded: styles.rounded,
					point: styles.point
				},
				compact: {
					false: styles.margin,
					true: styles.compact
				},
				underlined: {
					false: styles.hideHolder,
					true: null
				}
			}
		}), [styles])({
			compact,
			underlined: hasContent,
			variant
		}), className),
		items,
		...rest,
		classNames: {
			...rest?.classNames,
			popup: {
				root: styles.dropdown,
				...rest?.classNames?.popup
			}
		},
		more: {
			icon: /* @__PURE__ */ jsx(ActionIcon_default, { icon: MoreHorizontalIcon }),
			...rest?.more
		}
	});
};
Tabs$1.displayName = "Tabs";
var Tabs_default = Tabs$1;

//#endregion
export { Tabs_default as default };
