// @lobehub/ui/es/Accordion/AccordionItem.mjs
'use client';

import { useAccordionContext } from "./context.mjs";
import { useStyles } from "./style.mjs";
import FlexBasic_default from "../Flex/FlexBasic.mjs";
import Block_default from "../Block/Block.mjs";
import { LazyMotion } from "../motion/LazyMotion.mjs";
import { m } from "../motion/index.mjs";
import Text_default from "../Text/Text.mjs";
import ArrowIcon_default from "./ArrowIcon.mjs";
import { memo, useCallback, useMemo } from "react";
import { Fragment as Fragment$1, jsx, jsxs } from "react/jsx-runtime";

//#region src/Accordion/AccordionItem.tsx
const AccordionItem = memo(({ itemKey, title, children, action, disabled = false, hideIndicator: itemHideIndicator, indicatorPlacement: itemIndicatorPlacement, indicator: customIndicator, classNames, paddingInline = 16, paddingBlock = 8, padding, ref, variant: customVariant, styles: customStyles, headerWrapper }) => {
	const { cx, styles } = useStyles();
	const { isExpanded, onToggle, hideIndicator: contextHideIndicator, indicatorPlacement: contextIndicatorPlacement, keepContentMounted, disableAnimation, motionProps: contextMotionProps, variant = "filled" } = useAccordionContext();
	const isOpen = isExpanded(itemKey);
	const hideIndicatorFinal = itemHideIndicator ?? contextHideIndicator ?? false;
	const indicatorPlacementFinal = itemIndicatorPlacement ?? contextIndicatorPlacement ?? "start";
	const handleToggle = useCallback(() => {
		if (!disabled) onToggle(itemKey);
	}, [
		disabled,
		itemKey,
		onToggle
	]);
	const handleKeyDown = useCallback((e) => {
		if (disabled) return;
		switch (e.key) {
			case "Enter":
			case " ":
				e.preventDefault();
				handleToggle();
				break;
		}
	}, [disabled, handleToggle]);
	const indicator = useMemo(() => {
		if (hideIndicatorFinal) return null;
		if (customIndicator) {
			if (typeof customIndicator === "function") return /* @__PURE__ */ jsx("span", {
				"aria-hidden": "true",
				className: cx(styles.indicator, classNames?.indicator),
				style: customStyles?.indicator,
				children: customIndicator({
					isDisabled: disabled,
					isOpen
				})
			});
			return /* @__PURE__ */ jsx("span", {
				"aria-hidden": "true",
				className: cx(styles.indicator, classNames?.indicator),
				style: customStyles?.indicator,
				children: customIndicator
			});
		}
		return /* @__PURE__ */ jsx("span", {
			"aria-hidden": "true",
			className: cx(styles.indicator, classNames?.indicator),
			style: customStyles?.indicator,
			children: /* @__PURE__ */ jsx(ArrowIcon_default, { className: cx(styles.icon, isOpen && styles.iconRotate) })
		});
	}, [
		hideIndicatorFinal,
		customIndicator,
		disabled,
		isOpen,
		cx,
		styles,
		classNames,
		customStyles
	]);
	const motionProps = useMemo(() => ({
		animate: isOpen ? "enter" : "exit",
		exit: "exit",
		initial: false,
		variants: {
			enter: {
				height: "auto",
				opacity: 1,
				transition: {
					duration: .2,
					ease: [
						.4,
						0,
						.2,
						1
					]
				}
			},
			exit: {
				height: 0,
				opacity: 0,
				transition: {
					duration: .2,
					ease: [
						.4,
						0,
						.2,
						1
					]
				}
			}
		},
		...contextMotionProps
	}), [isOpen, contextMotionProps]);
	const contentElement = useMemo(() => {
		if (disableAnimation || !keepContentMounted) {
			if (keepContentMounted) return /* @__PURE__ */ jsx("div", {
				className: cx("accordion-content", styles.content, classNames?.content),
				role: "region",
				style: {
					display: isOpen ? "block" : "none",
					...customStyles?.content
				},
				children: /* @__PURE__ */ jsx("div", {
					className: styles.contentInner,
					children
				})
			});
			return isOpen && /* @__PURE__ */ jsx("div", {
				className: cx("accordion-content", styles.content, classNames?.content),
				role: "region",
				style: customStyles?.content,
				children: /* @__PURE__ */ jsx("div", {
					className: styles.contentInner,
					children
				})
			});
		}
		return /* @__PURE__ */ jsx(LazyMotion, { children: /* @__PURE__ */ jsx(m.div, {
			...motionProps,
			style: { overflow: "hidden" },
			children: /* @__PURE__ */ jsx("div", {
				className: cx("accordion-content", styles.content, classNames?.content),
				role: "region",
				style: customStyles?.content,
				children: /* @__PURE__ */ jsx("div", {
					className: styles.contentInner,
					children
				})
			})
		}) });
	}, [
		disableAnimation,
		keepContentMounted,
		isOpen,
		cx,
		styles,
		classNames,
		customStyles,
		children,
		motionProps
	]);
	const titleNode = useMemo(() => typeof title === "string" ? /* @__PURE__ */ jsx(Text_default, {
		className: classNames?.title,
		ellipsis: true,
		style: customStyles?.title,
		children: title
	}) : title, [
		title,
		classNames?.title,
		customStyles?.title
	]);
	const actionNode = useMemo(() => action && /* @__PURE__ */ jsx(FlexBasic_default, {
		align: "center",
		className: cx("accordion-action", styles.action, classNames?.action),
		flex: "none",
		gap: 4,
		horizontal: true,
		onClick: (e) => e.stopPropagation(),
		style: customStyles?.action,
		children: action
	}), [
		action,
		cx,
		styles,
		classNames?.action,
		customStyles?.action
	]);
	const headerElement = useMemo(() => {
		const header = /* @__PURE__ */ jsx(Block_default, {
			className: cx("accordion-header", styles.header, classNames?.header),
			clickable: !disabled,
			gap: 4,
			horizontal: true,
			justify: "space-between",
			onClick: handleToggle,
			onKeyDown: handleKeyDown,
			padding,
			paddingBlock,
			paddingInline,
			ref,
			style: {
				alignItems: "center",
				cursor: disabled ? "not-allowed" : "pointer",
				opacity: disabled ? .5 : void 0,
				overflow: "hidden",
				width: "100%",
				...customStyles?.header
			},
			variant: customVariant || variant,
			children: indicatorPlacementFinal === "start" ? /* @__PURE__ */ jsxs(Fragment$1, { children: [/* @__PURE__ */ jsxs(FlexBasic_default, {
				align: "center",
				flex: 1,
				gap: 2,
				horizontal: true,
				style: { overflow: "hidden" },
				children: [titleNode, indicator]
			}), /* @__PURE__ */ jsx(FlexBasic_default, {
				align: "center",
				flex: "none",
				gap: 4,
				horizontal: true,
				children: actionNode
			})] }) : /* @__PURE__ */ jsxs(Fragment$1, { children: [/* @__PURE__ */ jsx(FlexBasic_default, {
				align: "center",
				flex: 1,
				gap: 2,
				horizontal: true,
				style: { overflow: "hidden" },
				children: titleNode
			}), /* @__PURE__ */ jsxs(FlexBasic_default, {
				align: "center",
				flex: "none",
				gap: 4,
				horizontal: true,
				children: [actionNode, indicator]
			})] })
		});
		if (headerWrapper) return headerWrapper(header);
		return header;
	}, [
		cx,
		styles,
		classNames,
		disabled,
		handleToggle,
		handleKeyDown,
		padding,
		paddingBlock,
		paddingInline,
		ref,
		customVariant,
		variant,
		indicatorPlacementFinal,
		titleNode,
		indicator,
		actionNode,
		headerWrapper
	]);
	return /* @__PURE__ */ jsxs("div", {
		className: cx("accordion-item", styles.item, classNames?.base),
		style: customStyles?.base,
		children: [headerElement, contentElement]
	});
});
AccordionItem.displayName = "AccordionItem";
var AccordionItem_default = AccordionItem;

//#endregion
export { AccordionItem_default as default };
