// @lobehub/ui/es/Accordion/style.mjs
import { createStyles } from "antd-style";

//#region src/Accordion/style.ts
const useStyles = createStyles(({ css: css$1, token }) => {
	return {
		action: css$1`
      width: 0;
      opacity: 0;
      transition:
        opacity,
        width 0.2s ${token.motionEaseOut};
    `,
		base: css$1`
      display: flex;
      flex-direction: column;
      width: 100%;
    `,
		content: css$1`
      overflow: hidden;
    `,
		contentInner: css$1`
      /* Content wrapper for animation */
    `,
		divider: css$1`
      margin-block: 0;
    `,
		header: css$1`
      &:hover .accordion-action {
        width: auto;
        opacity: 1;
      }

      &:focus-visible {
        border-radius: ${token.borderRadiusLG}px;
        outline: 2px solid ${token.colorPrimary};
        outline-offset: 2px;
      }
    `,
		icon: css$1`
      transition: transform 200ms ${token.motionEaseOut};
    `,
		iconRotate: css$1`
      transform: rotate(90deg);
    `,
		indicator: css$1`
      display: flex;
      flex-shrink: 0;
      align-items: center;

      font-size: 18px;
      color: ${token.colorTextDescription};

      transition: transform 200ms ${token.motionEaseOut};
    `,
		item: css$1`
      position: relative;
      display: flex;
      flex-direction: column;
    `
	};
});

//#endregion
export { useStyles };
