// @lobehub/ui/es/GuideCard/style.mjs
import { createStyles } from "antd-style";

//#region src/GuideCard/style.ts
const useStyles = createStyles(({ cx: cx$1, css: css$1, stylish, token, isDarkMode }) => ({
	borderless: stylish.variantBorderlessWithoutHover,
	close: css$1`
    position: absolute;
    inset-block-start: 8px;
    inset-inline-end: 8px;
  `,
	content: css$1`
    padding: 16px;
  `,
	cover: css$1`
    align-self: center;
  `,
	desc: css$1`
    color: ${token.colorTextDescription};
  `,
	filled: cx$1(stylish.variantFilledWithoutHover, css$1`
      background: linear-gradient(
        to bottom,
        ${isDarkMode ? token.colorFillTertiary : token.colorFillQuaternary},
        ${isDarkMode ? token.colorFillQuaternary : token.colorFillTertiary}
      );
    `),
	outlined: stylish.variantOutlinedWithoutHover,
	root: css$1`
    position: relative;
    overflow: hidden;
    border-radius: ${token.borderRadiusLG}px;
  `,
	shadow: stylish.shadow,
	title: css$1`
    font-size: 16px;
    font-weight: bold;
  `
}));

//#endregion
export { useStyles };
