// @lobehub/ui/es/Avatar/Avatar.mjs
'use client';

import Center_default from "../Flex/Center.mjs";
import Icon_default from "../Icon/Icon.mjs";
import Img_default from "../Img/index.mjs";
import FluentEmoji_default from "../FluentEmoji/FluentEmoji.mjs";
import { useStyles } from "./style.mjs";
import { isValidElement, memo, useMemo } from "react";
import { jsx, jsxs } from "react/jsx-runtime";
import { Avatar } from "antd";
import { cva } from "class-variance-authority";
import { Loader2 } from "lucide-react";
import { readableColor } from "polished";
import { getEmoji } from "@lobehub/fluent-emoji";

//#region src/Avatar/Avatar.tsx
const Avatar$1 = memo(({ bordered, className, avatar, title, animation, borderedColor, size = 48, shape = "square", background, style, unoptimized, alt, variant = "borderless", shadow, loading, sliceText = true, emojiScaleWithBackground = true, ref, ...rest }) => {
	const isStringAvatar = typeof avatar === "string";
	const isDefaultAntAvatar = useMemo(() => Boolean(avatar && ([
		"/",
		"http",
		"data:"
	].some((index) => isStringAvatar && avatar.startsWith(index)) || isValidElement(avatar))), [avatar, isStringAvatar]);
	const emoji = useMemo(() => avatar && !isDefaultAntAvatar && isStringAvatar && getEmoji(avatar), [
		avatar,
		isStringAvatar,
		isDefaultAntAvatar
	]);
	const { styles, cx, theme } = useStyles();
	const variants = useMemo(() => cva(styles.root, {
		defaultVariants: {
			shadow: false,
			variant: "borderless"
		},
		variants: {
			variant: {
				filled: styles.filled,
				outlined: styles.outlined,
				borderless: styles.borderless
			},
			shadow: {
				false: null,
				true: styles.shadow
			}
		}
	}), [styles]);
	const text = String(isDefaultAntAvatar ? title : avatar);
	const imgAlt = alt || title || "avatar";
	const defualtAvatar = typeof avatar === "string" ? /* @__PURE__ */ jsx(Img_default, {
		alt: imgAlt,
		height: size,
		loading: "lazy",
		src: avatar,
		unoptimized,
		width: size
	}) : avatar;
	const hasBackground = background && background !== "transparent" && background !== "rgba(0,0,0,0)" && background !== null;
	const customAvatar = useMemo(() => emoji ? /* @__PURE__ */ jsx(FluentEmoji_default, {
		emoji,
		size: emojiScaleWithBackground ? hasBackground ? size * .85 : size : size * .85,
		type: animation ? "anim" : "3d",
		unoptimized
	}) : sliceText ? (text || title)?.toUpperCase().slice(0, 2) : (text || title)?.toUpperCase(), [
		animation,
		emoji,
		hasBackground,
		size,
		sliceText,
		text,
		title,
		unoptimized,
		emojiScaleWithBackground
	]);
	return /* @__PURE__ */ jsxs(Avatar, {
		alt: imgAlt,
		className: cx(variants({
			shadow,
			variant
		}), className),
		draggable: false,
		ref,
		shape,
		size,
		src: isDefaultAntAvatar ? defualtAvatar : void 0,
		style: {
			background: isDefaultAntAvatar || !!emoji ? background : background || theme.colorBorder,
			boxShadow: bordered ? `${theme.colorBgLayout} 0 0 0 2px, ${borderedColor || theme.colorTextTertiary} 0 0 0 4px` : void 0,
			color: readableColor(background || theme.colorBorder),
			cursor: rest?.onClick ? "pointer" : void 0,
			fontSize: size * (emoji ? .7 : .5),
			...style
		},
		...rest,
		children: [loading && /* @__PURE__ */ jsx(Center_default, {
			className: styles.loading,
			height: "100%",
			width: "100%",
			children: /* @__PURE__ */ jsx(Icon_default, {
				icon: Loader2,
				spin: true
			})
		}), !isDefaultAntAvatar && customAvatar]
	});
});
Avatar$1.displayName = "Avatar";
var Avatar_default = Avatar$1;

//#endregion
export { Avatar_default as default };
