// @lobehub/ui/es/AutoComplete/Select.mjs
'use client';

import { useStyles } from "./style.mjs";
import { memo, useMemo } from "react";
import { jsx } from "react/jsx-runtime";
import { AutoComplete } from "antd";
import { cva } from "class-variance-authority";

//#region src/AutoComplete/Select.tsx
const AutoComplete$1 = memo(({ variant, shadow, className, ...rest }) => {
	const { styles, cx, theme } = useStyles();
	return /* @__PURE__ */ jsx(AutoComplete, {
		className: cx(useMemo(() => cva(styles.root, {
			defaultVariants: { shadow: false },
			variants: {
				variant: {
					filled: styles.filled,
					outlined: styles.outlined,
					borderless: styles.borderless,
					underlined: null
				},
				shadow: {
					false: null,
					true: styles.shadow
				}
			}
		}), [styles])({
			shadow,
			variant: variant || (theme.isDarkMode ? "filled" : "outlined")
		}), className),
		variant: variant || (theme.isDarkMode ? "filled" : "outlined"),
		...rest
	});
});
AutoComplete$1.displayName = "AutoComplete";
var Select_default = AutoComplete$1;

//#endregion
export { Select_default as default };
