// @lobehub/ui/es/HotkeyInput/style.mjs
import { createStyles } from "antd-style";

//#region src/HotkeyInput/style.ts
const useStyles = createStyles(({ css: css$1, token, stylish }) => {
	return {
		borderless: stylish.variantBorderless,
		disabled: stylish.disabled,
		error: css$1`
      border: 1px solid ${token.colorError};
    `,
		errorText: css$1`
      font-size: 12px;
      color: ${token.colorError};
    `,
		filled: stylish.variantFilled,
		focused: css$1`
      background: ${token.colorFillSecondary} !important;
    `,
		hiddenInput: css$1`
      cursor: text;

      position: absolute;
      z-index: -1;
      inset-block-start: 0;
      inset-inline-start: 0;

      width: 100%;
      height: 100%;

      opacity: 0;
    `,
		outlined: stylish.variantOutlined,
		placeholder: css$1`
      color: ${token.colorTextDescription};
    `,
		root: css$1`
      cursor: pointer;

      position: relative;

      max-width: 100%;
      height: 36px;
      padding-block: 0;
      padding-inline: 12px;

      border-radius: ${token.borderRadius}px;
    `,
		shadow: stylish.shadow
	};
});

//#endregion
export { useStyles };
