// @lobehub/ui/es/Tooltip/TooltipFloating.mjs
'use client';

import FlexBasic_default from "../Flex/FlexBasic.mjs";
import Hotkey_default from "../Hotkey/Hotkey.mjs";
import { LazyMotion } from "../motion/LazyMotion.mjs";
import { AnimatePresence, m } from "../motion/index.mjs";
import { useStyles } from "./style.mjs";
import React, { useMemo } from "react";
import { jsx, jsxs } from "react/jsx-runtime";
import { FloatingArrow } from "@floating-ui/react";
import { useDebounce } from "ahooks";

//#region src/Tooltip/TooltipFloating.tsx
const TooltipFloating = ({ open, title, placement, floatingStyles, setFloating, floatingProps, arrow: arrow$1, arrowRef, context, hotkey, hotkeyProps, className, classNames, styles: styleProps, zIndex }) => {
	const { styles, cx } = useStyles();
	const transformOrigin = useMemo(() => {
		switch (String(placement || "top").split("-")[0]) {
			case "top": return "bottom center";
			case "bottom": return "top center";
			case "left": return "center right";
			case "right": return "center left";
			default: return "center";
		}
	}, [placement]);
	const hasTransform = useDebounce(Boolean(floatingStyles?.transform), {
		leading: false,
		wait: 16
	});
	return /* @__PURE__ */ jsx(LazyMotion, { children: /* @__PURE__ */ jsx(AnimatePresence, { children: open && title && /* @__PURE__ */ jsx("div", {
		className: cx(styles.tooltip, hasTransform && styles.tooltipLayout, classNames?.container, classNames?.root, className),
		ref: setFloating,
		role: "tooltip",
		style: styleProps?.root ? {
			...floatingStyles,
			zIndex,
			...styleProps.container,
			...styleProps.root
		} : {
			...floatingStyles,
			zIndex,
			...styleProps?.container
		},
		...floatingProps,
		children: /* @__PURE__ */ jsxs(m.div, {
			animate: {
				opacity: 1,
				scale: 1
			},
			exit: {
				opacity: 0,
				scale: .98
			},
			initial: {
				opacity: 0,
				scale: .96
			},
			style: { transformOrigin },
			transition: {
				duration: .12,
				ease: [
					.4,
					0,
					.2,
					1
				]
			},
			children: [/* @__PURE__ */ jsx("div", {
				className: cx(styles.content, classNames?.content),
				style: styleProps?.content,
				children: hotkey ? /* @__PURE__ */ jsxs(FlexBasic_default, {
					align: "center",
					gap: 8,
					horizontal: true,
					justify: "space-between",
					children: [/* @__PURE__ */ jsx("span", { children: title }), /* @__PURE__ */ jsx(Hotkey_default, {
						inverseTheme: true,
						keys: hotkey,
						...hotkeyProps
					})]
				}) : title
			}), arrow$1 && context && /* @__PURE__ */ jsx(FloatingArrow, {
				className: cx(styles.arrow, classNames?.arrow),
				context,
				height: 6,
				ref: arrowRef,
				style: styleProps?.arrow,
				width: 12
			})]
		})
	}, "tooltip") }) });
};
var TooltipFloating_default = TooltipFloating;

//#endregion
export { TooltipFloating_default as default };
