// @lobehub/ui/es/Form/components/FormSubmitFooter.mjs
'use client';

import FlexBasic_default from "../../Flex/FlexBasic.mjs";
import Icon_default from "../../Icon/Icon.mjs";
import Button_default from "../../Button/Button.mjs";
import { useSubmitFooterStyles } from "../style.mjs";
import { useFormContext } from "./FormProvider.mjs";
import { merge, removeUndefined } from "./merge.mjs";
import { memo, useEffect, useMemo, useState } from "react";
import { Fragment as Fragment$1, jsx, jsxs } from "react/jsx-runtime";
import { Form } from "antd";
import { motion } from "motion/react";
import { InfoIcon } from "lucide-react";
import isEqual from "fast-deep-equal";

//#region src/Form/components/FormSubmitFooter.tsx
const FormSubmitFooter = memo(({ enableReset = true, buttonProps, float, onReset, saveButtonProps, resetButtonProps, enableUnsavedWarning, children, texts, className, ...rest }) => {
	const { form, initialValues, submitLoading } = useFormContext();
	const [hasUnsavedChanges, setHasUnsavedChanges] = useState(false);
	const values = Form.useWatch([], form) || {};
	const { cx, styles, theme } = useSubmitFooterStyles();
	const v = useMemo(() => removeUndefined(values), [values]);
	const initialV = useMemo(() => removeUndefined(initialValues), [initialValues]);
	const mergedV = useMemo(() => merge(initialV, v), [v, initialV]);
	useEffect(() => {
		setHasUnsavedChanges(!isEqual(mergedV, initialV));
	}, [
		mergedV,
		initialV,
		submitLoading
	]);
	const fn = (e) => {
		if (hasUnsavedChanges) e.returnValue = texts?.unSavedWarning || "You have unsaved changes. Are you sure you want to leave?";
		else delete e.returnValue;
	};
	useEffect(() => {
		if (!enableUnsavedWarning) return;
		if (typeof window === "undefined" || !hasUnsavedChanges) return;
		window.addEventListener("beforeunload", fn);
		return () => window.removeEventListener("beforeunload", fn);
	}, [enableUnsavedWarning, hasUnsavedChanges]);
	const content = /* @__PURE__ */ jsxs(Fragment$1, { children: [
		(float || hasUnsavedChanges) && /* @__PURE__ */ jsxs(Fragment$1, { children: [/* @__PURE__ */ jsx(Icon_default, {
			color: theme.colorTextDescription,
			icon: InfoIcon,
			size: 12,
			style: { marginLeft: 8 }
		}), /* @__PURE__ */ jsx("span", {
			style: {
				color: theme.colorTextDescription,
				flex: "none",
				fontSize: 12,
				marginRight: float ? 16 : 4
			},
			children: texts?.unSaved || "Unsaved changes"
		})] }),
		children,
		enableReset && (float || hasUnsavedChanges) && /* @__PURE__ */ jsx(Button_default, {
			htmlType: "button",
			onClick: () => {
				onReset?.(v, initialV);
				form?.resetFields();
			},
			shape: float ? "round" : void 0,
			variant: "filled",
			...buttonProps,
			...resetButtonProps,
			children: texts?.reset || "Reset"
		}),
		/* @__PURE__ */ jsx(Button_default, {
			htmlType: "submit",
			loading: submitLoading,
			shape: float ? "round" : void 0,
			type: "primary",
			...buttonProps,
			...saveButtonProps,
			children: texts?.submit || "Submit"
		})
	] });
	if (!float) return /* @__PURE__ */ jsx(FlexBasic_default, {
		align: "center",
		className: cx(styles.footer, className),
		gap: 8,
		horizontal: true,
		justify: "flex-end",
		...rest,
		children: content
	});
	return /* @__PURE__ */ jsx(motion.div, {
		animate: hasUnsavedChanges ? "visible" : "hidden",
		className: styles.floatFooter,
		initial: "hidden",
		transition: {
			duration: .1,
			ease: "easeOut"
		},
		variants: {
			hidden: {
				opacity: 0,
				x: "-50%",
				y: 20
			},
			visible: {
				opacity: 1,
				x: "-50%",
				y: 0
			}
		},
		children: /* @__PURE__ */ jsx(FlexBasic_default, {
			align: "center",
			className,
			gap: 8,
			horizontal: true,
			justify: "center",
			...rest,
			children: content
		})
	});
});
FormSubmitFooter.displayName = "FormSubmitFooter";
var FormSubmitFooter_default = FormSubmitFooter;

//#endregion
export { FormSubmitFooter_default as default };
