// @lobehub/ui/es/Form/components/FormItem.mjs
'use client';

import { useItemStyles } from "../style.mjs";
import FormDivider_default from "./FormDivider.mjs";
import FormTitle_default from "./FormTitle.mjs";
import { memo, useMemo } from "react";
import { Fragment as Fragment$1, jsx, jsxs } from "react/jsx-runtime";
import { Form } from "antd";
import { cva } from "class-variance-authority";

//#region src/Form/components/FormItem.tsx
const { Item } = Form;
const FormItem = memo(({ desc, tag, minWidth, avatar, className, label, children, divider, layout, variant, ...rest }) => {
	const { cx, styles } = useItemStyles({ minWidth });
	const variants = useMemo(() => cva(styles.root, {
		defaultVariants: {
			divider: false,
			itemMinWidth: false,
			layout: "vertical"
		},
		variants: {
			itemMinWidth: {
				true: styles.itemMinWidth,
				false: null
			},
			divider: {
				true: null,
				false: styles.itemNoDivider
			},
			layout: {
				vertical: styles.verticalLayout,
				horizontal: null
			}
		}
	}), [styles]);
	return /* @__PURE__ */ jsxs(Fragment$1, { children: [divider && /* @__PURE__ */ jsx(FormDivider_default, { visible: variant !== "borderless" }), /* @__PURE__ */ jsx(Item, {
		className: cx(variants({
			divider,
			itemMinWidth: Boolean(minWidth),
			layout
		}), className),
		label: /* @__PURE__ */ jsx(FormTitle_default, {
			avatar,
			desc,
			tag,
			title: label
		}),
		layout,
		...rest,
		children
	})] });
});
FormItem.displayName = "FormItem";
var FormItem_default = FormItem;

//#endregion
export { FormItem_default as default };
