// @lobehub/ui/es/Form/components/FormTitle.mjs
'use client';

import FlexBasic_default from "../../Flex/FlexBasic.mjs";
import Tag_default from "../../Tag/Tag.mjs";
import { useTitleStyles } from "../style.mjs";
import { jsx, jsxs } from "react/jsx-runtime";

//#region src/Form/components/FormTitle.tsx
const FormTitle = ({ tag, title, desc, avatar, classNames, styles: customStyles, ...rest }) => {
	const { cx, styles } = useTitleStyles();
	return /* @__PURE__ */ jsxs(FlexBasic_default, {
		align: `center`,
		gap: 8,
		horizontal: true,
		...rest,
		children: [avatar, /* @__PURE__ */ jsxs(FlexBasic_default, {
			className: cx(styles.content, classNames?.content),
			gap: 8,
			style: customStyles?.content,
			children: [/* @__PURE__ */ jsxs(FlexBasic_default, {
				align: "center",
				className: cx(styles.title, classNames?.title),
				direction: "horizontal",
				gap: 8,
				style: customStyles?.title,
				children: [title, tag && /* @__PURE__ */ jsx(Tag_default, {
					className: classNames?.tag,
					style: customStyles?.tag,
					children: tag
				})]
			}), desc && /* @__PURE__ */ jsx("small", {
				className: cx(styles.desc, classNames?.desc),
				style: customStyles?.desc,
				children: desc
			})]
		})]
	});
};
FormTitle.displayName = "FormTitle";
var FormTitle_default = FormTitle;

//#endregion
export { FormTitle_default as default };
