// @lobehub/ui/es/Form/components/FormFlatGroup.mjs
'use client';

import FlexBasic_default from "../../Flex/FlexBasic.mjs";
import { useFlatGroupStyles } from "../style.mjs";
import { memo, useMemo } from "react";
import { jsx } from "react/jsx-runtime";
import { useResponsive } from "antd-style";
import { cva } from "class-variance-authority";

//#region src/Form/components/FormFlatGroup.tsx
const FormFlatGroup = memo(({ className, children, variant = "borderless", ...rest }) => {
	const { mobile } = useResponsive();
	const { cx: cx$1, styles } = useFlatGroupStyles(variant);
	const vriants = useMemo(() => cva(styles.root, {
		defaultVariants: { variant: "borderless" },
		variants: { variant: {
			filled: styles.filled,
			outlined: styles.outlined,
			borderless: styles.borderless
		} }
	}), [styles]);
	return /* @__PURE__ */ jsx(FlexBasic_default, {
		className: cx$1(mobile ? styles.mobile : vriants({ variant }), className),
		...rest,
		children
	});
});
FormFlatGroup.displayName = "FormFlatGroup";
var FormFlatGroup_default = FormFlatGroup;

//#endregion
export { FormFlatGroup_default as default };
