// @lobehub/ui/es/Form/components/FormGroup.mjs
'use client';

import FlexBasic_default from "../../Flex/FlexBasic.mjs";
import Icon_default from "../../Icon/Icon.mjs";
import Collapse_default from "../../Collapse/Collapse.mjs";
import { useGroupStyles } from "../style.mjs";
import { memo, useMemo } from "react";
import { jsx, jsxs } from "react/jsx-runtime";
import { useResponsive } from "antd-style";
import { cva } from "class-variance-authority";
import { isUndefined } from "lodash-es";

//#region src/Form/components/FormGroup.tsx
const FormGroup = memo(({ className, icon, title, children, extra, variant = "borderless", defaultActive = true, collapsible, active, keyValue = "group", onCollapse, desc, ...rest }) => {
	const { mobile } = useResponsive();
	const { cx: cx$1, styles } = useGroupStyles(variant);
	const isBorderless = variant === "borderless";
	const defaultCollapsible = isUndefined(collapsible) ? !isBorderless : collapsible;
	const titleContent = /* @__PURE__ */ jsxs(FlexBasic_default, {
		className: cx$1(useMemo(() => cva(styles.title, {
			defaultVariants: { variant: "borderless" },
			variants: { variant: {
				filled: null,
				outlined: null,
				borderless: styles.titleBorderless
			} }
		}), [styles])({ variant }), styles.titleMobile),
		gap: 8,
		horizontal: true,
		children: [icon && /* @__PURE__ */ jsx(Icon_default, { icon }), title]
	});
	if (mobile) return /* @__PURE__ */ jsxs(FlexBasic_default, {
		className,
		children: [/* @__PURE__ */ jsxs(FlexBasic_default, {
			className: styles.mobileGroupHeader,
			horizontal: true,
			justify: "space-between",
			children: [titleContent, extra]
		}), /* @__PURE__ */ jsx("div", {
			className: styles.mobileGroupBody,
			children
		})]
	});
	return /* @__PURE__ */ jsx(Collapse_default, {
		activeKey: isUndefined(active) ? void 0 : active ? [keyValue] : [],
		className,
		classNames: {
			header: variant === "borderless" ? styles.titleBorderless : void 0,
			title: variant === "borderless" ? styles.titleBorderless : void 0
		},
		collapsible: defaultCollapsible,
		defaultActiveKey: defaultActive ? [keyValue] : void 0,
		items: [{
			children,
			desc,
			extra,
			icon,
			key: keyValue,
			label: title
		}],
		onChange: (v) => onCollapse?.(v.length > 0),
		variant,
		...rest
	});
});
FormGroup.displayName = "FormGroup";
var FormGroup_default = FormGroup;

//#endregion
export { FormGroup_default as default };
