// @lobehub/ui/es/Select/Select.mjs
'use client';

import Icon_default from "../Icon/Icon.mjs";
import { useStyles } from "./style.mjs";
import { memo, useMemo } from "react";
import { jsx } from "react/jsx-runtime";
import { Select } from "antd";
import { cva } from "class-variance-authority";
import { ChevronDownIcon } from "lucide-react";

//#region src/Select/Select.tsx
const Input$1 = memo(({ ref, variant, suffixIconProps, suffixIcon, shadow, className, ...rest }) => {
	const { styles, cx, theme } = useStyles();
	return /* @__PURE__ */ jsx(Select, {
		className: cx(useMemo(() => cva(styles.root, {
			defaultVariants: { shadow: false },
			variants: {
				variant: {
					filled: styles.filled,
					outlined: styles.outlined,
					borderless: styles.borderless,
					underlined: null
				},
				shadow: {
					false: null,
					true: styles.shadow
				}
			}
		}), [styles])({
			shadow,
			variant: variant || (theme.isDarkMode ? "filled" : "outlined")
		}), className),
		ref,
		suffixIcon: /* @__PURE__ */ jsx(Icon_default, {
			icon: suffixIcon || ChevronDownIcon,
			size: "small",
			...suffixIconProps,
			style: {
				pointerEvents: "none",
				...suffixIconProps?.style
			}
		}),
		variant: variant || (theme.isDarkMode ? "filled" : "outlined"),
		...rest
	});
});
Input$1.displayName = "Input";
var Select_default = Input$1;

//#endregion
export { Select_default as default };
