// @lobehub/ui/es/Segmented/Segmented.mjs
'use client';

import { useStyles } from "./style.mjs";
import { memo, useMemo } from "react";
import { jsx } from "react/jsx-runtime";
import { Segmented } from "antd";
import { cva } from "class-variance-authority";

//#region src/Segmented/Segmented.tsx
const Segmented$1 = memo(({ ref, padding, style, className, variant = "filled", shadow, glass, ...rest }) => {
	const { cx, styles } = useStyles();
	return /* @__PURE__ */ jsx(Segmented, {
		className: cx(useMemo(() => cva(styles.root, {
			defaultVariants: {
				glass: false,
				shadow: false,
				variant: "filled"
			},
			variants: {
				variant: {
					filled: styles.filled,
					outlined: styles.outlined,
					borderless: styles.borderless
				},
				glass: {
					false: null,
					true: styles.glass
				},
				shadow: {
					false: null,
					true: styles.shadow
				}
			}
		}), [styles])({
			glass,
			shadow,
			variant
		}), className),
		ref,
		style: {
			padding,
			...style
		},
		...rest
	});
});
Segmented$1.displayName = "Segmented";
var Segmented_default = Segmented$1;

//#endregion
export { Segmented_default as default };
