// @lobehub/ui/es/storybook/StoryBook/index.mjs
'use client';

import FlexBasic_default from "../../Flex/FlexBasic.mjs";
import Center_default from "../../Flex/Center.mjs";
import DraggablePanel_default from "../../DraggablePanel/index.mjs";
import { useStyles } from "./style.mjs";
import { memo } from "react";
import { jsx, jsxs } from "react/jsx-runtime";
import { useResponsive } from "antd-style";
import { LevaPanel, useControls, useCreateStore } from "leva";

//#region src/storybook/StoryBook/index.tsx
const StoryBook = memo(({ ref, levaStore, noPadding, className, children, ...rest }) => {
	const { mobile } = useResponsive();
	const { styles, cx: cx$1 } = useStyles(Boolean(noPadding));
	return /* @__PURE__ */ jsxs(FlexBasic_default, {
		align: "stretch",
		className: cx$1(styles.editor, className),
		horizontal: !mobile,
		justify: "stretch",
		ref,
		children: [/* @__PURE__ */ jsx(Center_default, {
			className: styles.left,
			flex: 1,
			...rest,
			children
		}), /* @__PURE__ */ jsx(DraggablePanel_default, {
			className: styles.right,
			minWidth: 280,
			placement: mobile ? "bottom" : "right",
			children: /* @__PURE__ */ jsxs("div", {
				className: styles.leva,
				children: [/* @__PURE__ */ jsx(LevaPanel, {
					fill: true,
					flat: true,
					store: levaStore,
					titleBar: false
				}), " "]
			})
		})]
	});
});
StoryBook.displayName = "StoryBook";
var StoryBook_default = StoryBook;

//#endregion
export { StoryBook_default as default, useControls, useCreateStore };
