// @lobehub/ui/es/storybook/StoryBook/style.mjs
import { createStyles } from "antd-style";

//#region src/storybook/StoryBook/style.ts
const useStyles = createStyles(({ cx: cx$1, css: css$1, token, responsive }, noPadding) => {
	return {
		editor: css$1`
      width: inherit;
      min-height: inherit;
    `,
		left: cx$1(css$1`
        position: relative;
        overflow: auto;
      `, !noPadding && css$1`
          padding-block: 40px;
          padding-inline: 24px;
        `),
		leva: css$1`
      --leva-sizes-controlWidth: 66%;
      --leva-colors-elevation1: ${token.colorFillSecondary};
      --leva-colors-elevation2: transparent;
      --leva-colors-elevation3: ${token.colorFillSecondary};
      --leva-colors-accent1: ${token.colorPrimary};
      --leva-colors-accent2: ${token.colorPrimaryHover};
      --leva-colors-accent3: ${token.colorPrimaryActive};
      --leva-colors-highlight1: ${token.colorTextTertiary};
      --leva-colors-highlight2: ${token.colorTextSecondary};
      --leva-colors-highlight3: ${token.colorText};
      --leva-colors-vivid1: ${token.colorWarning};
      --leva-shadows-level1: unset;
      --leva-shadows-level2: unset;
      --leva-fonts-mono: ${token.fontFamilyCode};

      overflow: auto;

      width: 100%;
      height: 100%;
      padding-block: 6px;
      padding-inline: 0;

      > div {
        background: transparent;

        > div {
          background: transparent;
        }
      }

      input:checked + label > svg {
        stroke: ${token.colorBgLayout};
      }

      button {
        --leva-colors-accent2: ${token.colorFillSecondary};
      }
    `,
		right: css$1`
      background: ${token.colorBgLayout};

      ${responsive.mobile} {
        .draggable-panel-fixed {
          width: 100% !important;
        }
      }
    `
	};
});

//#endregion
export { useStyles };
