// @lobehub/ui/es/awesome/TypewriterEffect/style.mjs
import { createStyles } from "antd-style";

//#region src/awesome/TypewriterEffect/style.ts
const useStyles = createStyles(({ css: css$1, token, isDarkMode }) => ({
	container: css$1`
    display: inline-block;
    white-space: pre-wrap;
  `,
	cursor: css$1`
    transform: translateY(10%);

    display: inline-block;
    align-items: center;

    width: 3px;
    height: 1em;
    margin-inline-start: 0.25rem;
    border-radius: 2px;

    opacity: 1;
    background-color: ${token.colorPrimary};
  `,
	cursorBlock: css$1`
    transform: translateY(10%);

    display: inline-block;
    align-items: center;

    width: 0.5em;
    height: 1em;
    margin-inline-start: 0.25rem;
    border-radius: 2px;

    opacity: 1;
    background-color: ${token.colorPrimary};
  `,
	cursorCustom: css$1`
    display: inline-block;
    align-items: center;
    margin-inline-start: 0.25rem;
    opacity: 1;
  `,
	cursorDot: css$1`
    display: inline-block;
    align-items: center;

    width: 0.75em;
    height: 0.75em;
    margin-inline-start: 0.25rem;
    border-radius: 50%;

    opacity: 1;
    background-color: ${token.colorPrimary};
  `,
	cursorHidden: css$1`
    display: none;
  `,
	cursorUnderscore: css$1`
    transform: translateY(0.3em);

    display: inline-block;
    align-items: center;

    width: 0.6em;
    height: 0.15em;
    margin-inline-start: 0.25rem;
    border-radius: 2px;

    opacity: 1;
    background-color: ${token.colorPrimary};
  `,
	text: css$1`
    color: ${isDarkMode ? token.colorTextLightSolid : token.colorText};
  `
}));

//#endregion
export { useStyles };
