// @lobehub/ui/es/ThemeSwitch/ThemeSwitch.mjs
'use client';

import FlexBasic_default from "../Flex/FlexBasic.mjs";
import Icon_default from "../Icon/Icon.mjs";
import ActionIcon_default from "../ActionIcon/ActionIcon.mjs";
import Dropdown_default from "../Dropdown/Dropdown.mjs";
import { useMemo } from "react";
import { jsx, jsxs } from "react/jsx-runtime";
import { Select } from "antd";
import { Monitor, Moon, Sun } from "lucide-react";

//#region src/ThemeSwitch/ThemeSwitch.tsx
const DEFAULT_ICON_SET = {
	auto: Monitor,
	dark: Moon,
	light: Sun
};
const ThemeSwitch = ({ size = "middle", themeMode, onThemeSwitch, type = "icon", labels = {
	auto: "System",
	dark: "Dark",
	light: "Light"
}, className, variant, style }) => {
	const items = useMemo(() => [
		{
			icon: DEFAULT_ICON_SET.auto,
			key: "auto",
			label: labels.auto
		},
		{
			icon: DEFAULT_ICON_SET.light,
			key: "light",
			label: labels.light
		},
		{
			icon: DEFAULT_ICON_SET.dark,
			key: "dark",
			label: labels.dark
		}
	], [labels]);
	return type === "select" ? /* @__PURE__ */ jsx(Select, {
		className,
		defaultValue: themeMode,
		onChange: onThemeSwitch,
		options: items.map((item) => ({
			label: /* @__PURE__ */ jsxs(FlexBasic_default, {
				align: "center",
				gap: 8,
				horizontal: true,
				children: [/* @__PURE__ */ jsx(Icon_default, { icon: item.icon }), item.label]
			}),
			value: item.key
		})),
		style,
		variant
	}) : /* @__PURE__ */ jsx(Dropdown_default, {
		menu: {
			items,
			onClick: (e) => onThemeSwitch(e.key)
		},
		trigger: ["click"],
		children: /* @__PURE__ */ jsx(ActionIcon_default, {
			className,
			icon: DEFAULT_ICON_SET[themeMode],
			size,
			style,
			variant
		})
	});
};
ThemeSwitch.displayName = "ThemeSwitch";
var ThemeSwitch_default = ThemeSwitch;

//#endregion
export { ThemeSwitch_default as default };
