// @lobehub/ui/es/ColorSwatches/ColorSwatches.mjs
'use client';

import FlexBasic_default from "../Flex/FlexBasic.mjs";
import Center_default from "../Flex/Center.mjs";
import Icon_default from "../Icon/Icon.mjs";
import Tooltip_default from "../Tooltip/Tooltip.mjs";
import { useStyles } from "./style.mjs";
import { useMemo } from "react";
import { jsx, jsxs } from "react/jsx-runtime";
import { ColorPicker } from "antd";
import useMergeState from "use-merge-value";
import { CheckIcon } from "lucide-react";
import { readableColor, rgba } from "polished";
import chroma from "chroma-js";

//#region src/ColorSwatches/ColorSwatches.tsx
const ColorSwatches = ({ enableColorPicker, enableColorSwatches = true, defaultValue, value, style, colors, onChange, size = 24, shape = "circle", texts, ref, ...rest }) => {
	const [active, setActive] = useMergeState(defaultValue, {
		defaultValue,
		onChange,
		value
	});
	const { cx, styles, theme } = useStyles(size);
	const isCustomActive = useMemo(() => active && active !== theme.colorPrimary && !colors.some((c) => c.color === active), [
		active,
		colors,
		theme.colorPrimary
	]);
	return /* @__PURE__ */ jsxs(FlexBasic_default, {
		gap: 6,
		horizontal: true,
		ref,
		style: {
			flexWrap: "wrap",
			...style
		},
		...rest,
		children: [enableColorSwatches && colors.map((c, i) => {
			const color = c.color || theme.colorPrimary;
			const isActive = !active && !c.color || color === active;
			const isTransparent = c.color === "transparent" || chroma(c.color).alpha() === 0;
			return /* @__PURE__ */ jsx(Tooltip_default, {
				title: c.title,
				children: /* @__PURE__ */ jsx(Center_default, {
					className: cx(styles.container, isTransparent && styles.transparent, isActive && styles.active),
					onClick: () => setActive(c.color || void 0),
					style: {
						background: isTransparent ? void 0 : color,
						borderRadius: shape === "circle" ? "50%" : theme.borderRadius
					},
					children: isActive && /* @__PURE__ */ jsx(Icon_default, {
						color: rgba(readableColor(color), .33),
						icon: CheckIcon,
						size: {
							size: 14,
							strokeWidth: 4
						},
						style: { pointerEvents: "none" }
					})
				})
			}, c?.key || i);
		}), enableColorPicker && /* @__PURE__ */ jsx(Tooltip_default, {
			title: texts?.custom || "Custom",
			children: /* @__PURE__ */ jsx(ColorPicker, {
				arrow: false,
				className: cx(styles.picker, enableColorSwatches && styles.conic, isCustomActive && styles.active),
				defaultValue: theme.colorPrimary,
				disabledAlpha: true,
				format: "hex",
				onChangeComplete: (c) => {
					if (c.toHexString() === theme.colorPrimary) setActive("");
					else setActive(c.toHexString());
				},
				presets: enableColorSwatches ? void 0 : [{
					colors: colors.map((c) => c.color),
					label: texts?.presets || "Presets"
				}],
				style: { borderRadius: shape === "circle" ? "50%" : theme.borderRadius },
				value: enableColorSwatches ? void 0 : active
			})
		})]
	});
};
ColorSwatches.displayName = "ColorSwatches";
var ColorSwatches_default = ColorSwatches;

//#endregion
export { ColorSwatches_default as default };
