// @lobehub/ui/es/ColorSwatches/style.mjs
import { createStyles } from "antd-style";

//#region src/ColorSwatches/style.ts
const useStyles = createStyles(({ css: css$1, token, prefixCls }, size) => {
	return {
		active: css$1`
      box-shadow: inset 0 0 0 1px ${token.colorFill};
    `,
		conic: css$1`
      background: conic-gradient(
        ${token.red},
        ${token.volcano},
        ${token.orange},
        ${token.gold},
        ${token.yellow},
        ${token.lime},
        ${token.green},
        ${token.cyan},
        ${token.blue},
        ${token.geekblue},
        ${token.purple},
        ${token.magenta},
        ${token.red}
      );
      .${prefixCls}-color-picker-color-block {
        opacity: 0;
      }
    `,
		container: css$1`
      cursor: pointer;

      flex: none;

      width: ${size}px;
      min-width: ${size}px;
      height: ${size}px;
      min-height: ${size}px;

      background: ${token.colorBgContainer};
      box-shadow: inset 0 0 0 1px ${token.colorFillSecondary};

      &:hover {
        box-shadow:
          inset 0 0 0 1px rgba(0, 0, 0, 5%),
          0 0 0 2px ${token.colorText};
      }
    `,
		picker: css$1`
      overflow: hidden;
      flex: none;

      width: ${size}px;
      min-width: ${size}px;
      height: ${size}px;
      min-height: ${size}px;
      padding: 0;

      border: none;
      box-shadow: inset 0 0 0 1px ${token.colorFillSecondary};

      &:hover {
        box-shadow:
          inset 0 0 0 1px ${token.colorFillSecondary},
          0 0 0 2px ${token.colorText};
      }

      .${prefixCls}-color-picker-color-block {
        width: 100%;
        height: 100%;
        border: none;
        border-radius: inherit;
      }
    `,
		transparent: css$1`
      background-image: conic-gradient(
        ${token.colorFillSecondary} 25%,
        transparent 25% 50%,
        ${token.colorFillSecondary} 50% 75%,
        transparent 75% 100%
      );
      background-size: 50% 50%;
    `
	};
});

//#endregion
export { useStyles };
