// @lobehub/ui/es/awesome/Features/style.mjs
import { createStyles } from "antd-style";

//#region src/awesome/Features/style.ts
const useStyles = createStyles(({ token, prefixCls, css: css$1, cx: cx$1 }, { rowNum, hasLink }) => {
	const prefix = `${prefixCls}-features`;
	const coverCls = `${prefix}-cover`;
	const descCls = `${prefix}-description`;
	const titleCls = `${prefix}-title`;
	const imgCls = `${prefix}-img`;
	const scaleUnit = 20;
	const genSize = (size) => css$1`
      width: ${size}px;
      height: ${size}px;
      font-size: ${size * (22 / 24)}px;
    `;
	const withTransition = css$1`
      transition: all ${token.motionDurationSlow} ${token.motionEaseInOutCirc};
    `;
	return {
		cell: css$1`
        overflow: hidden;
      `,
		container: cx$1(withTransition, css$1`
          position: relative;
          z-index: 1;

          overflow: hidden;

          height: 228px;
          max-height: 228px;
          padding: 24px;

          p {
            font-size: 16px;
            line-height: 1.2;
            text-align: start;
            word-break: break-word;
          }

          &:hover {
            .${coverCls} {
              width: 100%;
              height: ${scaleUnit * rowNum}px;
              padding: 0;
              background: ${token.colorFillContent};
            }

            .${imgCls} {
              ${genSize(100)};
            }

            .${descCls} {
              position: absolute;
              visibility: hidden;
              opacity: 0;
            }

            .${titleCls} {
              font-size: ${hasLink ? 14 : 20}px;
            }
          }
        `),
		desc: cx$1(descCls, withTransition, css$1`
          pointer-events: none;
          color: ${token.colorTextSecondary};

          quotient {
            position: relative;

            display: block;

            margin-block: 12px;
            margin-inline: 0;
            padding-inline-start: 12px;

            color: ${token.colorTextDescription};
          }
        `),
		img: cx$1(imgCls, withTransition, css$1`
          ${genSize(20)};
          color: ${token.colorText};
        `),
		imgContainer: cx$1(coverCls, withTransition, css$1`
          ${genSize(24)};
          padding: 4px;
          opacity: 0.8;
          border-radius: ${token.borderRadius}px;
        `),
		link: css$1`
        ${withTransition};
        margin-block-start: 24px;
      `,
		title: cx$1(titleCls, withTransition, css$1`
          pointer-events: none;

          margin-block: 16px;
          margin-inline: 0;

          font-size: 20px;
          line-height: ${token.lineHeightHeading3};
          color: ${token.colorText};
        `)
	};
});

//#endregion
export { useStyles };
