// @lobehub/ui/es/Tag/Tag.mjs
'use client';

import { colorsPreset, colorsPresetSystem, presetColors, presetSystemColors } from "./utils.mjs";
import { useStyles } from "./styles.mjs";
import { useMemo } from "react";
import { jsx } from "react/jsx-runtime";
import { Tag } from "antd";
import { cva } from "class-variance-authority";

//#region src/Tag/Tag.tsx
const Tag$1 = ({ className, ref, size = "middle", color, variant = "filled", children, onClick, style, ...rest }) => {
	const { styles, cx, theme } = useStyles();
	const variants = useMemo(() => cva(styles.root, {
		defaultVariants: {
			size: "middle",
			variant: "filled"
		},
		variants: {
			variant: {
				filled: styles.filled,
				outlined: styles.outlined,
				borderless: styles.borderless
			},
			size: {
				small: styles.small,
				middle: null,
				large: styles.large
			}
		}
	}), [styles]);
	const colors = useMemo(() => {
		let textColor = theme.colorTextSecondary;
		let backgroundColor;
		let borderColor;
		const isBorderless = variant === "borderless";
		const isFilled = variant === "filled";
		const isPresetColor = color && presetColors.includes(color);
		const isPresetSystemColors = color && presetSystemColors.has(color);
		const isHexColor = color && color.startsWith("#");
		if (isPresetColor) {
			textColor = colorsPreset(theme, color);
			backgroundColor = isBorderless ? "transparent" : colorsPreset(theme, color, "fillTertiary");
			borderColor = colorsPreset(theme, color, isFilled ? "fillQuaternary" : "fillTertiary");
		}
		if (isPresetSystemColors) {
			textColor = colorsPresetSystem(theme, color);
			backgroundColor = isBorderless ? "transparent" : colorsPresetSystem(theme, color, "fillTertiary");
			borderColor = colorsPresetSystem(theme, color, isFilled ? "fillQuaternary" : "fillTertiary");
		}
		if (isHexColor) {
			textColor = theme.colorBgLayout;
			backgroundColor = isBorderless ? "transparent" : color;
		}
		return {
			backgroundColor,
			borderColor,
			textColor
		};
	}, [
		color,
		theme,
		variant
	]);
	return /* @__PURE__ */ jsx(Tag, {
		className: cx(variants({
			size,
			variant
		}), className),
		color,
		onClick,
		ref,
		style: {
			background: colors?.backgroundColor,
			borderColor: colors?.borderColor,
			color: colors?.textColor,
			cursor: onClick ? "pointer" : void 0,
			...style
		},
		variant: variant === "borderless" ? "outlined" : variant,
		...rest,
		children
	});
};
Tag$1.displayName = "Tag";
var Tag_default = Tag$1;

//#endregion
export { Tag_default as default };
