// @lobehub/ui/es/mobile/TabBar/style.mjs
import { createStyles } from "antd-style";

//#region src/mobile/TabBar/style.ts
const useStyles = createStyles(({ css: css$1, token }) => {
	return {
		active: css$1`
      color: ${token.colorPrimary};
    `,
		container: css$1`
      user-select: none;

      overflow: hidden;
      flex: none;

      width: 100vw;

      background: ${token.colorBgLayout};
      border-block-start: 1px solid ${token.colorFillTertiary};
    `,
		icon: css$1`
      width: 24px;
      height: 24px;
      font-size: 24px;
    `,
		inner: css$1`
      position: relative;
      width: 100%;
      height: 48px;
    `,
		tab: css$1`
      cursor: pointer;
      width: 48px;
      height: 48px;
      color: ${token.colorTextDescription};
    `,
		title: css$1`
      overflow: hidden;

      width: 100%;
      margin-block-start: -0.125em;

      font-size: 12px;
      line-height: 1.125em;
      text-align: center;
      text-overflow: ellipsis;
      white-space: nowrap;
    `
	};
});

//#endregion
export { useStyles };
