// @lobehub/ui/es/mobile/ChatHeader/ChatHeaderTitle.mjs
'use client';

import FlexBasic_default from "../../Flex/FlexBasic.mjs";
import { useTitleStyles } from "./style.mjs";
import { memo } from "react";
import { jsx, jsxs } from "react/jsx-runtime";

//#region src/mobile/ChatHeader/ChatHeaderTitle.tsx
const ChatHeaderTitle = memo(({ title, desc, tag }) => {
	const { styles } = useTitleStyles();
	if (desc) return /* @__PURE__ */ jsxs(FlexBasic_default, {
		align: "center",
		flex: 1,
		gap: 4,
		justify: "center",
		children: [/* @__PURE__ */ jsxs(FlexBasic_default, {
			align: "center",
			className: styles.titleContainer,
			gap: 4,
			horizontal: true,
			children: [/* @__PURE__ */ jsx("div", {
				className: styles.titleWithDesc,
				children: title
			}), tag && /* @__PURE__ */ jsx(FlexBasic_default, {
				className: styles.tag,
				horizontal: true,
				children: tag
			})]
		}), /* @__PURE__ */ jsx(FlexBasic_default, {
			align: "center",
			horizontal: true,
			children: /* @__PURE__ */ jsx("div", {
				className: styles.desc,
				children: desc
			})
		})]
	});
	return /* @__PURE__ */ jsxs(FlexBasic_default, {
		align: "center",
		flex: 1,
		gap: 4,
		horizontal: true,
		justify: "center",
		children: [/* @__PURE__ */ jsx("div", {
			className: styles.title,
			children: title
		}), /* @__PURE__ */ jsx(FlexBasic_default, {
			className: styles.tag,
			horizontal: true,
			children: tag
		})]
	});
});
ChatHeaderTitle.displayName = "ChatHeaderTitle";
var ChatHeaderTitle_default = ChatHeaderTitle;

//#endregion
export { ChatHeaderTitle_default as default };
