// @lobehub/ui/es/mobile/ChatHeader/style.mjs
import { createStyles } from "antd-style";

//#region src/mobile/ChatHeader/style.ts
const useStyles = createStyles(({ css: css$1, token }) => {
	return {
		center: css$1`
      height: 100%;
    `,
		container: css$1`
      overflow: hidden;
      flex: none;
      width: 100vw;
      background: ${token.colorBgLayout};
    `,
		inner: css$1`
      position: relative;

      width: 100%;
      height: 44px;
      min-height: 44px;
      max-height: 44px;
      padding-block: 0;
      padding-inline: 6px;
    `,
		left: css$1`
      justify-content: flex-start;
      height: 100%;
    `,
		right: css$1`
      justify-content: flex-end;
      height: 100%;
    `
	};
});
const useTitleStyles = createStyles(({ css: css$1, token }) => ({
	desc: css$1`
    overflow: hidden;

    width: 100%;

    font-size: 12px;
    line-height: 1;
    color: ${token.colorTextTertiary};
    text-overflow: ellipsis;
    white-space: nowrap;
  `,
	tag: css$1`
    flex: none;
  `,
	title: css$1`
    overflow: hidden;

    font-size: 16px;
    font-weight: bold;
    line-height: 1;
    text-overflow: ellipsis;
    white-space: nowrap;
  `,
	titleContainer: css$1`
    flex: 1;
  `,
	titleWithDesc: css$1`
    overflow: hidden;

    font-weight: bold;
    line-height: 1;
    text-overflow: ellipsis;
    white-space: nowrap;
  `
}));

//#endregion
export { useStyles, useTitleStyles };
