// @lobehub/ui/es/SideNav/SideNav.mjs
'use client';

import FlexBasic_default from "../Flex/FlexBasic.mjs";
import { useStyles } from "./style.mjs";
import { memo } from "react";
import { jsx, jsxs } from "react/jsx-runtime";

//#region src/SideNav/SideNav.tsx
const SideNav = memo(({ className, avatar, topActions, bottomActions, ...rest }) => {
	const { styles, cx } = useStyles();
	return /* @__PURE__ */ jsxs(FlexBasic_default, {
		align: "center",
		className: cx(styles, className),
		flex: "none",
		justify: "space-between",
		...rest,
		children: [/* @__PURE__ */ jsxs(FlexBasic_default, {
			align: "center",
			direction: "vertical",
			gap: 12,
			children: [avatar, /* @__PURE__ */ jsx(FlexBasic_default, {
				align: "center",
				direction: "vertical",
				gap: 8,
				children: topActions
			})]
		}), /* @__PURE__ */ jsx(FlexBasic_default, {
			align: "center",
			direction: "vertical",
			gap: 4,
			children: bottomActions
		})]
	});
});
SideNav.displayName = "SideNav";
var SideNav_default = SideNav;

//#endregion
export { SideNav_default as default };
