// @lobehub/ui/es/mdx/mdxComponents/Citation/index.mjs
'use client';

import PopoverPanel_default from "./PopoverPanel.mjs";
import { jsx } from "react/jsx-runtime";
import { createStyles } from "antd-style";
import { isEmpty } from "lodash-es";

//#region src/mdx/mdxComponents/Citation/index.tsx
const useStyles = createStyles(({ css: css$1 }) => ({
	container: css$1`
    display: inline-flex;
    line-height: var(--lobe-markdown-line-height);
    vertical-align: baseline;

    a {
      color: inherit;
    }
  `,
	supContainer: css$1`
    vertical-align: super;
  `
}));
const Citation = ({ children, href, inSup, id, citationDetail }) => {
	const { styles, cx: cx$1 } = useStyles();
	const usePopover = !isEmpty(citationDetail);
	const url = citationDetail?.url || href;
	if (inSup) return /* @__PURE__ */ jsx(PopoverPanel_default, {
		...citationDetail,
		usePopover,
		children: /* @__PURE__ */ jsx("span", {
			className: styles.container,
			children: /* @__PURE__ */ jsx("a", {
				"aria-describedby": "footnote-label",
				"data-footnote-ref": "true",
				href: url,
				id,
				rel: "noreferrer",
				target: citationDetail?.url ? "_blank" : void 0,
				children
			})
		})
	});
	return /* @__PURE__ */ jsx(PopoverPanel_default, {
		...citationDetail,
		usePopover,
		children: /* @__PURE__ */ jsx("sup", {
			className: cx$1(styles.container, styles.supContainer),
			children: url ? /* @__PURE__ */ jsx("a", {
				"aria-describedby": "footnote-label",
				"data-footnote-ref": true,
				href: url,
				rel: "noreferrer",
				target: "_blank",
				children
			}) : /* @__PURE__ */ jsx("span", {
				"aria-describedby": "footnote-label",
				"data-footnote-ref": true,
				children
			})
		})
	});
};
Citation.displayName = "MdxCitation";
var Citation_default = Citation;

//#endregion
export { Citation_default as default };
