// @lobehub/ui/es/DatePicker/DatePicker.mjs
'use client';

import { useStyles } from "./style.mjs";
import { memo, useMemo } from "react";
import { jsx } from "react/jsx-runtime";
import { DatePicker } from "antd";
import { cva } from "class-variance-authority";

//#region src/DatePicker/DatePicker.tsx
const DatePicker$1 = memo(({ variant, shadow, className, ...rest }) => {
	const { styles, cx, theme } = useStyles();
	return /* @__PURE__ */ jsx(DatePicker, {
		className: cx(useMemo(() => cva(styles.root, {
			defaultVariants: { shadow: false },
			variants: {
				variant: {
					filled: styles.filled,
					outlined: styles.outlined,
					borderless: styles.borderless,
					underlined: null
				},
				shadow: {
					false: null,
					true: styles.shadow
				}
			}
		}), [styles])({
			shadow,
			variant: variant || (theme.isDarkMode ? "filled" : "outlined")
		}), className),
		variant: variant || (theme.isDarkMode ? "filled" : "outlined"),
		...rest
	});
});
DatePicker$1.displayName = "DatePicker";
var DatePicker_default = DatePicker$1;

//#endregion
export { DatePicker_default as default };
