// @lobehub/ui/es/Hotkey/style.mjs
import { createStyles } from "antd-style";
import { rgba } from "polished";

//#region src/Hotkey/style.ts
const useStyles = createStyles(({ cx: cx$1, css: css$1, token, isDarkMode, stylish }) => {
	return {
		borderless: cx$1(stylish.variantBorderlessWithoutHover, css$1`
        padding-inline: 4px;
      `),
		filled: stylish.variantFilledWithoutHover,
		inverseTheme: css$1`
      color: ${rgba(token.colorBgContainer, .75)};
      background: ${rgba(token.colorBgContainer, isDarkMode ? .08 : .16)};
    `,
		outlined: stylish.variantOutlinedWithoutHover,
		root: css$1`
      overflow: hidden;

      min-width: 1.8em;
      height: 1.8em;
      padding-block: 0;
      padding-inline: 8px;

      font-family: ${token.fontFamily};
      font-size: 12px;
      line-height: 1.1;
      color: ${token.colorTextSecondary};
      text-align: center;
      white-space: nowrap;

      border: none;
      border-radius: ${token.borderRadiusSM}px;
    `
	};
});

//#endregion
export { useStyles };
