// @lobehub/ui/es/Highlighter/SyntaxHighlighter/index.mjs
'use client';

import StaticRenderer_default from "./StaticRenderer.mjs";
import { useStyles } from "./style.mjs";
import StreamRenderer_default from "./StreamRenderer.mjs";
import { memo, useMemo } from "react";
import { jsx } from "react/jsx-runtime";
import { cva } from "class-variance-authority";

//#region src/Highlighter/SyntaxHighlighter/index.tsx
const SyntaxHighlighter = memo(({ animated, children, className, enableTransformer, language, style, theme, variant = "borderless" }) => {
	const { styles, cx } = useStyles();
	const isDefaultTheme = theme === "lobe-theme" || !theme;
	const showBackground = !isDefaultTheme && variant === "filled";
	const resolvedTheme = isDefaultTheme ? void 0 : theme;
	const variants = useMemo(() => cva(styles.root, {
		defaultVariants: {
			shiki: true,
			showBackground: false,
			variant: "borderless"
		},
		variants: {
			shiki: {
				false: styles.unshiki,
				true: styles.shiki
			},
			variant: {
				filled: styles.padding,
				outlined: styles.padding,
				borderless: styles.noPadding
			},
			showBackground: {
				false: styles.noBackground,
				true: null
			}
		}
	}), [styles]);
	const shikiClassName = cx(variants({
		shiki: true,
		showBackground,
		variant
	}), className);
	const fallbackClassName = cx(variants({
		shiki: false,
		showBackground,
		variant
	}), className);
	if (animated) return /* @__PURE__ */ jsx(StreamRenderer_default, {
		className: shikiClassName,
		enableTransformer,
		fallbackClassName,
		language,
		style,
		theme: resolvedTheme,
		children
	});
	return /* @__PURE__ */ jsx(StaticRenderer_default, {
		className: shikiClassName,
		enableTransformer,
		fallbackClassName,
		language,
		style,
		theme: resolvedTheme,
		children
	});
}, (prevProps, nextProps) => prevProps.children === nextProps.children && prevProps.language === nextProps.language);
SyntaxHighlighter.displayName = "SyntaxHighlighter";
var SyntaxHighlighter_default = SyntaxHighlighter;

//#endregion
export { SyntaxHighlighter_default as default };
