// @lobehub/ui/es/Highlighter/SyntaxHighlighter/StreamRenderer.mjs
'use client';

import { useHighlight } from "../../hooks/useHighlight.mjs";
import { useStyles } from "./style.mjs";
import { memo } from "react";
import { jsx } from "react/jsx-runtime";
import { getTokenStyleObject } from "@shikijs/core";

//#region src/Highlighter/SyntaxHighlighter/StreamRenderer.tsx
const applyColorReplacement = (color, replacements) => {
	if (!color || !replacements) return color;
	return replacements[color.toLowerCase()] || color;
};
const normalizeStyleKeys = (style) => {
	const normalized = {};
	Object.entries(style).forEach(([key, value]) => {
		const normalizedKey = key.replaceAll(/-([a-z])/g, (_, char) => char.toUpperCase());
		normalized[normalizedKey] = value;
	});
	return normalized;
};
const getTokenInlineStyle = (token, replacements) => {
	const baseStyle = normalizeStyleKeys(token.htmlStyle || getTokenStyleObject(token));
	if (!replacements) return {
		...baseStyle,
		whiteSpace: "pre"
	};
	const style = {
		...baseStyle,
		whiteSpace: "pre"
	};
	if (style.color && typeof style.color === "string") style.color = applyColorReplacement(style.color, replacements);
	if (style.backgroundColor && typeof style.backgroundColor === "string") style.backgroundColor = applyColorReplacement(style.backgroundColor, replacements);
	return style;
};
const TokenSpan = memo(({ token, replacements }) => {
	return /* @__PURE__ */ jsx("span", {
		style: getTokenInlineStyle(token, replacements),
		children: token.content
	}, token.content);
}, (prev, next) => prev.token === next.token);
const TokenLine = memo(({ line, replacements }) => {
	if (!line.length) return /* @__PURE__ */ jsx("span", {
		className: "line",
		children: /* @__PURE__ */ jsx("span", {
			style: { whiteSpace: "pre" },
			children: "\xA0"
		})
	});
	return /* @__PURE__ */ jsx("span", {
		className: "line",
		children: line.map((token, tokenIndex) => /* @__PURE__ */ jsx(TokenSpan, {
			replacements,
			token
		}, `token-${tokenIndex}`))
	});
}, (prev, next) => prev.line === next.line);
const StreamRenderer = memo(({ children, className, enableTransformer, fallbackClassName, language, style, theme }) => {
	const { cx } = useStyles();
	const safeChildren = children ?? "";
	const { streaming } = useHighlight(safeChildren, {
		enableTransformer,
		language,
		streaming: true,
		theme
	});
	const lines = streaming?.lines;
	const preStyle = streaming?.preStyle;
	const replacements = streaming?.colorReplacements;
	if (!lines || lines.length === 0) return /* @__PURE__ */ jsx("div", {
		className: fallbackClassName,
		dir: "ltr",
		style,
		children: /* @__PURE__ */ jsx("pre", { children: /* @__PURE__ */ jsx("code", { children: safeChildren }) })
	});
	return /* @__PURE__ */ jsx("div", {
		className,
		dir: "ltr",
		style,
		children: /* @__PURE__ */ jsx("pre", {
			className: cx("shiki", theme),
			style: preStyle,
			tabIndex: 0,
			children: /* @__PURE__ */ jsx("code", {
				style: {
					display: "flex",
					flexDirection: "column",
					whiteSpace: "pre"
				},
				children: lines.map((line, index) => /* @__PURE__ */ jsx(TokenLine, {
					line,
					replacements
				}, `line-${index}`))
			})
		})
	});
});
StreamRenderer.displayName = "StreamRenderer";
var StreamRenderer_default = StreamRenderer;

//#endregion
export { StreamRenderer_default as default };
