// @lobehub/ui/es/Skeleton/SkeletonButton.mjs
'use client';

import { useStyles } from "./style.mjs";
import SkeletonBlock_default from "./SkeletonBlock.mjs";
import { jsx } from "react/jsx-runtime";

//#region src/Skeleton/SkeletonButton.tsx
const HEIGHT_MAP = {
	default: 36,
	large: 46,
	small: 28
};
const SkeletonButton = ({ active, block = false, shape = "default", size = "default", width, height, style, className, ...rest }) => {
	const { theme } = useStyles();
	const resolvedSize = size ?? "default";
	const baseHeight = height ?? HEIGHT_MAP[resolvedSize];
	const finalWidth = width ?? (block ? "100%" : shape === "circle" ? baseHeight : 80);
	const RADIUS_MAP = {
		default: theme.borderRadius,
		large: theme.borderRadiusLG,
		small: theme.borderRadiusSM
	};
	return /* @__PURE__ */ jsx(SkeletonBlock_default, {
		active,
		className,
		height: baseHeight,
		style: {
			borderRadius: shape === "circle" ? "50%" : shape === "round" ? `${theme.borderRadius * 2}px` : RADIUS_MAP[resolvedSize],
			...style
		},
		width: finalWidth,
		...rest
	});
};
SkeletonButton.displayName = "SkeletonButton";
var SkeletonButton_default = SkeletonButton;

//#endregion
export { SkeletonButton_default as default };
